# PHD3



Two protein lists:

* Hypoxia inducible factor 1alpha signaling pathway: CYBB HIF1A MIR210 PDK1 PDK3 RWDD3
* PHD3 interactome: ACC1 ACC2 FAR1 PRKDC MSH6 VPS4A XPO1 SLC25A12 YME1L1 MLF2 SLC17A4 ACSL3 SLC25A22 FAF2 SBDS CCT6B



# 0. Prepare analysis

Check Docker Version

```bash
dockerd
docker version
```

`Note` If you got error: Is the docker daemon running?

```bash
sudo systemctl start docker
sudo service docker start
```

Download kisudsoe/postgwas:latest image

```bash
docker pull kisudsoe/postgwas
```

Download `db_gwas.zip` file

gdown, Ref: https://medium.com/@acpanjan/download-google-drive-files-using-wget-3c2c025a8b99

```bash
yum -y install python-pip
pip install gdown --user
gdown https://drive.google.com/uc?id=1D6CVKi7-83a-g6Gd8X0UhVrYHCcph-wl
```

Unzip the file

```bash
mkdir db_gwas
unzip -d db_gwas db_gwas.zip
```



# 1. Gene expression by age

Prepare input file `util-GTEx_age/20.08-PHD3/input_22.tsv`

Run `--age` function to get age-correlated gene list:

```CMD
Rscript gtex-exe.r --age --gene 20.08-PHD3/input_22.tsv --dbage gtex_age_list_bonf_0.05.rds --out 20.08-PHD3
```

> ** Function age start..
> Read, 20.08-PHD3/input_22.tsv = [1] 22  4
> Read, gtex_age_list_bonf_0.05.rds = [1] 49
>
>   Extract gene lists.. done
>
> Figure draw: 20.08-PHD3/venn_input_22_Age-correlated genes_Bonf0.05.png
>
>   Extract Ensgids = [1] 22
>   Overlap the genes with GTEx Age DB:
>
>   1 Adipose_Subcutaneous... [1] 4 8
>   2 Adipose_Visceral_(Omentum)... [1] 8 8
>   3 Adrenal_Gland... NULL
>   4 Artery_Aorta... [1] 5 8
>   5 Artery_Coronary... NULL
>   6 Artery_Tibial... [1] 10  8
>   7 Bladder... NULL
>   8 Brain_Amygdala... NULL
>   9 Brain_Anterior_cingulate_cortex_(BA24)... NULL
>   10 Brain_Caudate_(basal_ganglia)... NULL
>   11 Brain_Cerebellar_Hemisphere... [1] 1 8
>   12 Brain_Cerebellum... [1] 2 8
>   13 Brain_Cortex... NULL
>   14 Brain_Frontal_Cortex_(BA9)... NULL
>   15 Brain_Hippocampus... [1] 1 8
>   16 Brain_Hypothalamus... [1] 2 8
>   17 Brain_Nucleus_accumbens_(basal_ganglia)... NULL
>   18 Brain_Putamen_(basal_ganglia)... NULL
>   19 Brain_Spinal_cord_(cervical_c-1)... NULL
>   20 Brain_Substantia_nigra... NULL
>   21 Breast_Mammary_Tissue... [1] 1 8
>   22 Cells_Cultured fibroblasts... NULL
>   23 Cervix_Ectocervix... NULL
>   24 Colon_Sigmoid... [1] 4 8
>   25 Colon_Transverse... [1] 1 8
>   26 Esophagus_Gastroesophageal_Junction... [1] 6 8
>   27 Esophagus_Mucosa... [1] 9 8
>   28 Esophagus_Muscularis... [1] 3 8
>   29 Heart_Atrial_Appendage... NULL
>   30 Heart_Left_Ventricle... [1] 1 8
>   31 Liver... NULL
>   32 Lung... [1] 3 8
>   33 Minor_Salivary_Gland... NULL
>   34 Muscle_Skeletal... [1] 3 8
>   35 Nerve_Tibial... [1] 2 8
>   36 Ovary... NULL
>   37 Pancreas... NULL
>   38 Pituitary... NULL
>   39 Prostate... NULL
>   40 Skin_Not_Sun_Exposed_(Suprapubic)... [1] 3 8
>   41 Skin_Sun_Exposed_(Lower_leg)... [1] 2 8
>   42 Small_Intestine_Terminal_Ileum... NULL
>   43 Spleen... NULL
>   44 Stomach... NULL
>   45 Testis... NULL
>   46 Thyroid... [1] 2 8
>   47 Uterus... NULL
>   48 Vagina... [1] 1 8
>   49 Whole_Blood... [1] 12  8
>   Removing NULL from GTEx = [1] 23
>   Merge annotations = [1] 86 11
> Write TSV: 20.08-PHD3/age_bonf0.05_ann.tsv
> Write RDS: 20.08-PHD3/age_bonf0.05_ann.rds
> Job done: 2020-08-21 14:32:49 for 0.9 sec



Run `--hist` function to draw gene number histogram by tissues:

```CMD
Rscript gtex-exe.r --hist --geneage 20.08-PHD3/age_bonf0.05_ann.rds --out 20.08-PHD3
```

> ** Function age_histo start..
> Read, 20.08-PHD3/age_bonf0.05_ann.rds = [1] 86 11
>   Gene n = [1] 22
>   Preparing histogram dataset... count tissue.. [1] 23
>   Write PNG: 20.08-PHD3/hist_bonf0.05.png
>
> Job done: 2020-08-21 14:52:41 for 1.2 sec

Run `--heat` function to draw slope heatmap by tissues:

`ISSUE` Error to save image.. why? See `GTEx heatmap.ipynb` and `gtex-exe.r`

```CMD
Rscript gtex-exe.r --heat --geneage 20.08-PHD3/age_bonf0.05_ann.rds --out 20.08-PHD3
```

> ** Function age_heat start..
> Read, 20.08-PHD3/age_bonf0.05_ann.rds = [1] 86 11
>   Tissue common/specific genes:
>   Organ_n Gene_n
> 8      10      1
> 7       8      1
> 6       6      2
> 5       5      4
> 4       4      2
> 3       3      7
> 2       2      2
> 1       1      3
>   Write TSV: 20.08-PHD3/age_heatmap_bonf0.05.tsv
>
> Count Gene-Tissue numbers = [1] 86 12
>   Gene N = 22
>
> Organs_n = 23
>   Paste Gene name and Engsid... done
>   Spread Tissue-Age.Coef = [1] 22 25
>   Add organ list.. done
>   Write RDS: 20.08-PHD3/heatmap_bonf0.05_beta.rds
>
> Prepare to matrix...
>   reorder.. row.. col.. tissue..
>   direction.. split.. convert.. done
>   heatmap dim = [1] 22 23
>   Draw PNG file: 20.08-PHD3/beta_organ_bonf0.05_over_1.png
>
> Job done: 2020-08-21 16:56:59 for 1.8 sec



# 2. Query with gene list

Prepare input file `20.08-PHD3/input_22.tsv` by below format:

> <Symbol> <Name> <Ensgid>

mkdir data collecting folder

```bash
mkdir 20.08-PHD3
```

Upload query file

```CMD
scp "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3\20.08-PHD3\input_22.tsv" ec2-user@3.14.252.130:/ukb/20.08-PHD3
```

Run docker image for gene search in gwas & phewas database

```bash
docker run --rm -v '/ukb:/data' \
	kisudsoe/postgwas Rscript postgwas-exe.r \
	--utils gene \
	--base  /data/20.08-PHD3/input_22.tsv \
	--trait /data/db_gwas/gwas_phewas_ensgid_rsid_trait_pval_fill.tsv \
	--out   /data/20.08-PHD3
```

> ** Run function: utils.r/bash_script... Done
> Read, /ukb/20.08-PHD3/input_22.tsv = [1] 22  4
> Read, /ukb/db_gwas/gwas_phewas_ensgid_rsid_trait_pval_fill.tsv = [1] 534812      6
>   Query 22 ensgids, found # = [1] 152   6
> Write TSV file: /ukb/20.08-PHD3/input_snp_71.tsv
> Job done: 2020-08-20 03:09:36 for 2.4 sec

Download result file

```CMD
scp ec2-user@3.14.252.130:/ukb/20.08-PHD3/input_snp_71.tsv "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3\20.08-PHD3"
```



# 3. Candidate SNPs



## Seed SNPs - hg19 BED

Prepare input file by below format that includes `Rsid` column:

> <...> <Rsid> <...>

Run `--dbdown genebed` function:

```bash
docker run --rm -v '/ukb:/data' \
	kisudsoe/postgwas Rscript postgwas-exe.r \
	--dbdown genebed \
	--base   /data/20.08-PHD3/input_snp_71.tsv \
	--out    /data/20.08-PHD3 \
	--hg     hg19
```

> ** Run function: db_download.r/gene_bed...ready
> Read, /ukb/20.08-PHD3/input_snp_71.tsv = [1] 152   6
> Search biomart for SNP coordinates:
>   Query SNPs            = [1] 152
>   Hg19 result table     = [1] 71  4
>   Cytoband annotation... 71.. done
>   Merged table          = [1] 71  5
>
> Write TSV file: /ukb/20.08-PHD3/gwas_biomart_71.tsv
> Write BED file: /ukb/20.08-PHD3/gwas_biomart_71.bed

Download the result BED file and check any blanks: 

```CMD
scp ec2-user@3.14.252.130:/ukb/20.08-PHD3/gwas_biomart_71.tsv "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3\20.08-PHD3"
```

`NOTE` If there are any blanks, you have to fill them by manually.



## Incorporate LD linked SNPs (R<sup>2</sup>>0.6)

1000 Genome project populations:

| Population            | Subpopulations              |
| --------------------- | --------------------------- |
| AFR African           | YRI LWK GWD MSL ESN ASW ACB |
| AMR Ad Mixed American | MXL PUR CLM PEL             |
| EAS East Asian        | CHB JPT CHS CDX KHV         |
| EUR European          | CEU TSI FIN GBR IBS         |
| SAS South Asian       | GIH PJL BEB STU ITU         |

Download from LD link DB by `--ldink down` function:

```txt
# This code have bug...
docker run --rm -v '/ukb:/data' \
	kisudsoe/postgwas Rscript postgwas-exe.r \
	--ldlink down \
	--base   /data/20.08-PHD3/input_snp_71.tsv \
	--out    /data/20.08-PHD3/ldlink \
	--popul  AFR AMR EAS EUR SAS
```

`NOTE` When you run this function in Windows, there is invalid cross-device link bug. To avoid the bug, you need to run this alternative code:

```bash
docker run --rm -v '/ukb:/data' \
  kisudsoe/postgwas /bin/sh -c \
  "Rscript postgwas-exe.r --ldlink down --base /data/20.08-PHD3/input_snp_71.tsv --out /ldlink --popul AFR AMR EAS EUR SAS && mkdir /data/20.08-PHD3/ldlink && mv /ldlink/* /data/20.08-PHD3/ldlink"
  
# check the download
ls /ukb/20.08-PHD3/ldlink
```

> ** Run function ldlink_down... ready
> Rsid query = 71.. done
>   Files are moved to target folder:     /ukb/20.08-PHD3/ldlink
> Job done: 2020-08-20 04:09:51 for 32.7 min

Download LD link

```CMD
scp ec2-user@3.14.252.130:/ukb/20.08-PHD3/ldlink/* "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3\20.08-PHD3\ldlink"
```

`NOTE` For debugging biomaRt (hg38), run this code:

```CMD
docker run --rm -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" \
  kisudsoe/postgwas Rscript postgwas-exe.r \
  --ldlink chkbiomart \
  --out    /data/20.08-PHD3
```

Filtering LD data by R2 >0.6

`ERROR` If you get server connection error for hg38, try to change the host url by using  `--mirror useast/uswest/asia/www.ensembl.org`.

```bash
docker run --rm -v "/ukb:/data" \
	kisudsoe/postgwas Rscript postgwas-exe.r \
	--ldlink filter \
	--base   /data/20.08-PHD3/input_snp_71.tsv \
	--ldpath /data/20.08-PHD3/ldlink \
	--out    /data/20.08-PHD3 \
	--r2     0.6 \
	--hg     hg19
```

> ** Run function ldlink_filter...
> Read download files... 71
>   Read LDlink results           = [1] 106907     12
>     Filtering by "r2 > 0.6": [1] 3175   12
>     [Message] No filter criteria for Dprime.
>   Filtered data dimension       = [1] 3175    3
>   Excluded no rsid elements     = [1] 13
>
> Basic summary of LDlink results:
>   SNP Tier 1                    = 71
>   SNP Tier 2                    = 1498
>   SNP candidates                = 1569
>   SNP source annotation table   = [1] 1569    2
>
> Add annotations:
>   Calculate LD block index... [1] 50
>
> Search biomart for SNP coordinates:
>   Query SNPs            = [1] 1569
>   Hg19 result table     = [1] 1550    4
>   Cytoband annotation... 1569.. done
>   Merged table          = [1] 1569    8
>
> Write file: /data/20.08-PHD3/gwas_biomart_1569.tsv
> Job done: 2020-08-20 17:51:32 for 14.8 sec

Download the result file to local

```CMD
scp ec2-user@3.14.252.130:/ukb/20.08-PHD3/gwas_biomart_1569.tsv "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3\20.08-PHD3"
```



## Prepare input hg19 BED file

`NOTE` You should fill the NA coordinates manually in the file before you go next step. Then upload the file to server:

```CMD
scp "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3\20.08-PHD3\gwas_biomart_fill_1569.tsv" ec2-user@3.14.252.130:/ukb/20.08-PHD3
```

Run `--ldlink bed` function:

```bash
docker run --rm -v "/ukb:/data" \
  kisudsoe/postgwas Rscript postgwas-exe.r \
  --ldlink  bed \
  --base    /data/20.08-PHD3/gwas_biomart_fill_1569.tsv \
  --out     /data/20.08-PHD3
```

> ** Run function ldlink_bed...
> Read, /data/20.08-PHD3/gwas_biomart_fill_1569.tsv = [1] 1569    8
> Write file:     /data/20.08-PHD3/gwas_hg19_biomart_1569.bed
>
> Job done: 2020-08-20 18:09:49 for 1 sec





# 4. Functional annotations

Quick run by `ann-20.08-PHD3.sh` file in local:

```bash
docker run --rm -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" kisudsoe/postgwas /bin/sh -c "bash /data/ann-20.08-PHD3.sh"
```

Check log files and debugging:

```CMD
docker run --rm -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" kisudsoe/postgwas /bin/sh -c "bash /data/ann-test.sh"
```



---

Upload the bash file to AWS server:

```CMD
scp ann-20.08-PHD3.sh ec2-user@3.14.252.130:/ukb
```

In AWS server,

```bash
docker run --rm -v "/ukb:/data" kisudsoe/postgwas /bin/sh -c "bash /data/ann-20.08-PHD3.sh"
```

---

`NOTE` GTEx eQTL genes needs >10 GB RAM size, work in work station:

```bash
Rscript postgwas-exe.r ^
  --dbfilt gtex_ovl ^
  --base   20.08-PHD3/gwas_biomart_1569.bed ^
  --gtex   db_gwas/gtex_signif_5e-08_Ensgid_dt.rds ^
  --out    20.08-PHD3
```

> ** Run function: db_filter.r/gtex_overlap...
>   Your memory size = [1] 44.83
> ready
> Input GWAS SNPs N = 1569
> Read, gtex_signif_5e-08_Ensgid_dt.rds = [1] 30613850        7
>   Overlapped eQTL-gene pairs = [1] 43236
>   eQTLs N = [1] 1159
>   Associated eGenes = [1] 0
>
> Write file: 20.08-PHD3/gtex_signif_1159.tsv
> Generating BED files at "20.08-PHD3/gtex_eqtl" for 49 tissues.. done
>
> Job done: 2020-08-20 19:11:47 for 3.5 min

---



Get Union lists

```CMD
docker run --rm -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" kisudsoe/postgwas /bin/sh -c "bash /data/ann-20.08-PHD3.sh"
```

Check log files



# 5. Summary table

```CMD
docker run --rm -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" kisudsoe/postgwas /bin/sh -c "bash /data/summ-20.08-PHD3.sh"
```

> ** Run function: db_venn.r/summ_ann... ready
> 58 Files/folders input.
>   1 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Adipose_Subcutaneous_974.bed
>   2 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Adipose_Visceral_Omentum_748.bed
>   3 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Adrenal_Gland_467.bed
>   4 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Artery_Aorta_709.bed
>   5 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Artery_Coronary_564.bed
>   6 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Artery_Tibial_856.bed
>   7 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Brain_Amygdala_50.bed
>   8 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Brain_Anterior_cingulate_cortex_BA24_334.bed
>   9 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Brain_Caudate_basal_ganglia_313.bed
>   10 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Brain_Cerebellar_Hemisphere_566.bed
>   11 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Brain_Cerebellum_659.bed
>   12 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Brain_Cortex_447.bed
>   13 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Brain_Frontal_Cortex_BA9_464.bed
>   14 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Brain_Hippocampus_307.bed
>   15 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Brain_Hypothalamus_126.bed
>   16 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Brain_Nucleus_accumbens_basal_ganglia_460.bed
>   17 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Brain_Putamen_basal_ganglia_323.bed
>   18 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Brain_Spinal_cord_cervical_c-1_197.bed
>   19 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Brain_Substantia_nigra_63.bed
>   20 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Breast_Mammary_Tissue_779.bed
>   21 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Cells_Cultured_fibroblasts_839.bed
>   22 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Cells_EBV-transformed_lymphocytes_152.bed
>   23 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Colon_Sigmoid_515.bed
>   24 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Colon_Transverse_604.bed
>   25 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Esophagus_Gastroesophageal_Junction_673.bed
>   26 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Esophagus_Mucosa_783.bed
>   27 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Esophagus_Muscularis_816.bed
>   28 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Heart_Atrial_Appendage_700.bed
>   29 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Heart_Left_Ventricle_629.bed
>   30 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Kidney_Cortex_24.bed
>   31 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Liver_391.bed
>   32 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Lung_856.bed
>   33 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Minor_Salivary_Gland_340.bed
>   34 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Muscle_Skeletal_798.bed
>   35 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Nerve_Tibial_751.bed
>   36 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Ovary_453.bed
>   37 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Pancreas_548.bed
>   38 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Pituitary_446.bed
>   39 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Prostate_557.bed
>   40 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Skin_Not_Sun_Exposed_Suprapubic_795.bed
>   41 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Skin_Sun_Exposed_Lower_leg_849.bed
>   42 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Small_Intestine_Terminal_Ileum_264.bed
>   43 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Spleen_530.bed
>   44 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Stomach_570.bed
>   45 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Testis_747.bed
>   46 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Thyroid_1031.bed
>   47 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Uterus_119.bed
>   48 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Vagina_457.bed
>   49 /data/20.08-PHD3/gtex_eqtl/snp_gtex_Whole_Blood_709.bed
>   50 /data/20.08-PHD3/gtex_eqtl/snp_union_gtex_eqtl_1159.bed
>   51 /data/20.08-PHD3/summary/snp_encode_tfbs_195.bed
>   52 /data/20.08-PHD3/summary/snp_lncrnasnp_45.bed
>   53 /data/20.08-PHD3/summary/snp_regulome2b_51.bed
>   54 /data/20.08-PHD3/summary/snp_roadmap_enh_merge_245.bed
>   55 /data/20.08-PHD3/summary/snp_ucsc_cds_25.bed
>   56 /data/20.08-PHD3/summary/snp_ucsc_proximalPromoter_200.bed
>   57 /data/20.08-PHD3/summary/snp_ucsc_wholeGene_1211.bed
>   58 /data/20.08-PHD3/summary/snp_union_roadmap_over_245.bed
> Total 58 file(s) is/are input.
>
> ** Run function: db_venn.r/venn_bed...
>   Read 58 files
>
> [Message] Can't plot Venn diagram for more than 5 sets.
>
>
> [Message] Can't plot Euler plot.
> ** Back to function: db_venn.r/summ...
>   Returned union list dim = [1] 1506   62
>   [PASS] uni_save = FALSE
>
>   ENCODE dim = [1] 2574   13
>   Merge dim = [1] 1192   61
>   Extract snp-tfbs pair = [1] 195   2
>   Write a CSV file: /data/20.08-PHD3/summary_encode.csv
>
>   Nearest gene dim = [1] 1747    9
>   Search biomaRt... 77.. parsing.. 69.. done
>   Sub-merge = [1] 1747    5
>   Extract snp-gene pair = [1] 1569    2
>   Merge dim = [1] 1747   63
>   Write a CSV file: /data/20.08-PHD3/summary_nearest.csv
>
>   GTEx dim = [1] 43236     7
>   Search biomaRt... 60.. parsing.. 59.. done
>   Merge dim = [1] 43236    67
>   Extract snp-eGene pair = [1] 1159    2
>   Write a CSV file: /data/20.08-PHD3/summary_gtex.csv
>
>   GWAS dim = [1] 1569    8
>   Merge dim = [1] 1569   69
>   Write a CSV file: /data/20.08-PHD3/summary_gwas.csv
>
> Job done: 2020-08-21 02:35:36 for 26.6 sec



## Venn analysis

Enhancer SNPs from Roadmap & ENCODE

```CMD
docker run --rm ^
  -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" ^
  kisudsoe/postgwas Rscript /src/venn_analysis.r --fig ^
  --base /data/20.08-PHD3/summary/snp_encode_tfbs_195.bed /data/20.08-PHD3/summary/snp_roadmap_enh_merge_245.bed ^
  --out  /data/20.08-PHD3
```

> Figure draw: /data/20.08-PHD3/venn_snp_encode_tfbs_195_snp_roadmap_enh_merge_245.png

Generate Enhancer SNPs BED file.

Enhancer SNPs & RegulomeDB & GTEx eQTLs

```CMD
docker run --rm ^
  -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" ^
  kisudsoe/postgwas Rscript /src/venn_analysis.r --fig ^
  --base /data/20.08-PHD3/Enhancer-TFBS_53.bed /data/20.08-PHD3/summary/snp_regulome2b_51.bed /data/20.08-PHD3/gtex_eqtl/snp_union_gtex_eqtl_1159.bed ^
  --out  /data/20.08-PHD3
```

> ​      original fitted residuals regionError
>
> A            7      7         0       0.000
> B            3      3         0       0.000
> C         1078   1078         0       0.002
> A&B          2      0         2       0.002
> A&C         35     35         0       0.000
> B&C         37     37         0       0.000
> A&B&C        9      9         0       0.000
>
> diagError: 0.002
> stress:    0
>
> Euler fit is done.
>
> Figure draw: /data/20.08-PHD3/venn_Enhancer-TFBS_53_snp_regulome2b_51.png



## SNP-nexus

Ref: https://www.snp-nexus.org/v4/7aa26925/



## SNPsea

Download and unzip the ready-to-use program:

* [util-snpsea](https://drive.google.com/file/d/1EMlk-hmK6mulFYIq7TYaOqZWyvRcP6MB/view?usp=sharing)

```bash
wget https://drive.google.com/file/d/1EMlk-hmK6mulFYIq7TYaOqZWyvRcP6MB/view?usp=sharing
mkdir snpsea
unzip util-snpsea.zip -d snpsea
```

Prepare GWAS file `20.08-PHD3_1569.gwas`  by following below format:

> <CHR> <POS> <SNP>
>
> chr1   10000   rs100

Prepare bash file `20.08-PHD3.sh` to run SNPsea.

```bash
#!/usr/bin/env bash

options=(
    --snps              gwas/20.08-PHD3_1569.gwas
    --gene-matrix       FANTOM2014.gct.gz
    --gene-intervals    NCBIgenes2013.bed.gz
    --snp-intervals     TGP2011.bed.gz
    --null-snps         Lango2010.txt.gz
    --out               out/20.08-PHD3-FANTOM
    --slop              10e3
    --threads           8
    --null-snpsets      0
    --min-observations  100
    --max-iterations    1e7
)

# Run SNPsea.
./bin/snpsea ${options[*]}

# Create a horizontal bar plot of condition p-values.
./bin/snpsea-barplot out/20.08-PHD3-FANTOM


options=(
    --snps              gwas/20.08-PHD3_1569.gwas
    --gene-matrix       GeneAtlas2004.gct.gz
    --gene-intervals    NCBIgenes2013.bed.gz
    --snp-intervals     TGP2011.bed.gz
    --null-snps         Lango2010.txt.gz
    --out               out/20.08-PHD3-GA2004
    --slop              10e3
    --threads           8
    --null-snpsets      0
    --min-observations  100
    --max-iterations    1e7
)

# Run SNPsea.
./bin/snpsea ${options[*]}

# Create a horizontal bar plot of condition p-values.
./bin/snpsea-barplot out/20.08-PHD3-GA2004


options=(
    --snps              gwas/20.08-PHD3_1569.gwas
    --gene-matrix       GTEx_v8_gene_median_tpm.gct.gz
    --gene-intervals    Ensemblgenes2020.bed.gz
    --snp-intervals     TGP2011.bed.gz
    --null-snps         Lango2010.txt.gz
    --out               out/20.08-PHD3-GTEx
    --slop              10e3
    --threads           8
    --null-snpsets      0
    --min-observations  100
    --max-iterations    1e7
)

# Run SNPsea.
./bin/snpsea ${options[*]}

# Create a horizontal bar plot of condition p-values.
./bin/snpsea-barplot out/20.08-PHD3-GTEx
```

Run SNPsea through image

```CMD
docker run -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" kisudsoe/postgwas /bin/sh -c "cd /data/util-snpsea && bash ./20.08-PHD3.sh"
```

`ISSUE` GTEx "condition_pvalues.txt" file have error on last row `Whole` -> `Whole Blood`. Then Run snpsea-barplot:

```CMD
docker run -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" kisudsoe/postgwas /bin/sh -c "cd /data/util-snpsea && ./bin/snpsea-barplot out/20.08-PHD3-GTEx"
```

Change figure size for GTEx at `bin/snpsea-barplot`:

```python
width = float(20) # 10 (GA, FANTOM) -> 20 (gtex)
top = int(60)     # 20 (GA, FANTOM) -> 60 (gtex)
```

Run again:

```bash
docker run -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" kisudsoe/postgwas /bin/sh -c "cd /data/util-snpsea && ./bin/snpsea-barplot out/20.08-PHD3-GTEx"
```



## LD block region plot

LocusZoom, Ref: http://locuszoom.org/

GWAS Catalog, Ref: https://www.ebi.ac.uk/gwas/variants/rs75418188

UCSC browser hg18 <- Good for specific region with Hapmap data!





# -----

# . Debugging

```CMD
docker run --rm -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" ^
  kisudsoe/postgwas Rscript postgwas-exe.r ^
  --ldlink  bed ^
  --base    /data/20.08-PHD3/gwas_biomart_fill_1569.tsv ^
  --out     /data/20.08-PHD3

docker run --rm -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" ^
  kisudsoe/postgwas /bin/sh -c ^
  "Rscript postgwas-exe.r --utils bash --base /data/20.08-PHD3/gwas_biomart_1569.bed --out /data/20.08-PHD3 --ann /data/db_gwas && mv dist_20.08-PHD3.sh /data/dist_20.08-PHD3.sh && bash /data/dist_20.08-PHD3.sh"
```



`Development` Update the program source code

```CMD
docker run -it -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" ^
  kisudsoe/postgwas /bin/bash
```

```bash
cp -r /data/src /data/postgwas-exe.r ./
```

```CMD
docker ps
```

```CMD
docker stop 2b6f42b36b29
docker commit -a "jjy" 2b6f42b36b29 kisudsoe/postgwas:latest
```

Upload new image to Docker Hub

```bash
# v7-2020-08-20
docker tag kisudsoe/postgwas:latest kisudsoe/postgwas:7
docker login
docker push kisudsoe/postgwas:latest
docker push kisudsoe/postgwas:7
```

Download new image to AWS server

```bash
docker pull kisudsoe/postgwas
```

