# SLC16A11

Solute Carrier Family 16 Member 11

ENSG00000174326



# a. Run Postgwas Docker image

## Install Docker Toolbox

Docker Toolbox, Ref: https://docs.docker.com/toolbox/toolbox_install_windows/

Run Docker Quickstart Terminal:

```bash
$ dockerd
$ docker version
```



## Download image

Pull latest postgwas image `Postgwas:latest`.

```bash
$ docker pull kisudsoe/postgwas
```



## Mount local volume to container

Mount volume, Ref: https://headsigned.com/posts/mounting-docker-volumes-with-docker-toolbox-for-windows/

In Oracle VM Virtual Box Manager -> Settings (default machine) -> Shared Folders -> Add new shared folder.

* Don't forget to check `Auto-mount` and `Make Permanent` options

Transient Folders: `C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3` path named as `Postgwas_v3`

```bash
$ docker-machine restart # First time only
```



## Start up the Postgwas image

Run the image with mount volume:

```bash
docker run -it -v \
	"/Postgwas_v3:/postgwas/data" \
	kisudsoe/postgwas:latest \
	/bin/bash
```

Source code located in `/postgwas` directory and local directory mounted as `/source`



## Check Postgwas tool

This process run in the image: `root@738df134f971:/#`

Check running the function.

```bash
cd /postgwas
Rscript postgwas-exe.r --help
```



### (For devlopment) Update source codes

```bash
container$ cp -r /postgwas/data/src /postgwas/data/postgwas-exe.r /postgwas/
```



# b. Preparing GWAS and Phewas Traits

Run function --utils gene_traits

```CMD
Rscript postgwas-exe.r ^
  --utils genetraits ^
  --gwas   db_gwas/gwas_catalog/gwas_catalog_slim_r2020-07-14.tsv ^
  --phewas db_gwas/phewas/phewas-catalog.csv ^
  --age    db_gwas/gwas_phewas_age_traits.tsv ^
  --out    db_gwas
```

> Read, db_gwas/gwas_catalog/gwas_catalog_slim_r2020-07-14.tsv = [1] 189811     29
> Read, db_gwas/phewas/phewas-catalog.csv = [1] 215107      9
> Read, db_gwas/gwas_phewas_age_traits.tsv = [1] 533   6
>   gwas selection... 6.. [1] 269562      4
>   phewas selection = [1] 215107      4
>
>   search gwas ensgid biomaRt hg38 = [1] 31040     2
>   Among queried 31037, unfound Ensgid # = [1] 9049
>   Merge search result = [1] 269580      5
>
>   search phewas symbol biomaRt hg38 = [1] 1946    2
>   Among queried 1776, unfound Symbol # = [1] 147
>   Merge search result = [1] 265724      5
>
>   Bind gwas and phewas = [1] 535304      5
>   Merge age-realted traits = [1] 535304      6
> Write TSV file, db_gwas/gwas_phewas_ensgid_rsid_trait_pval.tsv
> Job done: 2020-08-02 01:52:52 for 1.9 min



# -------

# 1. Query with gene list

Input file: `/data/20.08_SLC16A11/SLC16A11.tsv`

Prepare input file by below format:

```txt
<Symbol> <Name> <Ensgid>
```

Search the input Ensgids:

```bash
Rscript postgwas-exe.r \
	--utils gene \
	--base  data/20.08_SLC16A11/SLC16A11.tsv \
	--trait data/db_gwas/gwas_phewas_ensgid_rsid_trait_pval_fill.tsv \
	--out   data/20.08_SLC16A11
```

> ** Run function: utils.r/bash_script... Done
> Read, data/20.08_SLC16A11/SLC16A11.tsv = [1] 1 3
> Read, data/db_gwas/gwas_phewas_ensgid_rsid_trait_pval_fill.tsv = [1] 534812      6
>   Query 1 ensgids, found # = [1] 6 6
> Write TSV file: data/20.08_SLC16A11/input_snp_6.tsv
> Job done: 2020-08-02 17:56:46 for 4.5 sec



# 2. Prepareing Candidate SNPs

## Seed SNPs - BED (hg19)

Prepare input file by below format that includes 'Rsid' column:

```txt
<...> <Rsid> <...>
```

Run --dbdown genebed function:

```bash
Rscript postgwas-exe.r \
	--dbdown genebed \
	--base   data/20.08_SLC16A11/input_snp_6.tsv \
	--out    data/20.08_SLC16A11 \
	--hg     hg19
```

> ** Run function: db_download.r/gene_bed...ready
> Read, data/20.08_SLC16A11/input_snp_6.tsv = [1] 6 6
> Search biomart for SNP coordinates:
>   Query SNPs            = [1] 6
>   Hg19 result table     = Cache found
> [1] 6 4
>   Cytoband annotation... 6.. done
>   Merged table          = [1] 6 5
>
> Write TSV file: data/20.08_SLC16A11/gwas_biomart_6.tsv
> Write BED file: data/20.08_SLC16A11/gwas_biomart_6.bed
> Job done: 2020-08-02 18:19:20 for 5.7 sec



## Incorporate LD linked SNPs (R2 >0.6)

1000 Genome project populations:

| Population            | Subpopulations              |
| --------------------- | --------------------------- |
| AFR African           | YRI LWK GWD MSL ESN ASW ACB |
| AMR Ad Mixed American | MXL PUR CLM PEL             |
| EAS East Asian        | CHB JPT CHS CDX KHV         |
| EUR European          | CEU TSI FIN GBR IBS         |
| SAS South Asian       | GIH PJL BEB STU ITU         |

Select populations from 6 studies of SLC16A11

* [GCST002317](https://www.ebi.ac.uk/gwas/studies/GCST002317), [type II diabetes mellitus](https://www.ebi.ac.uk/gwas/efotraits/EFO_0001360), 8,214 Hispanic or Latin American, 5648 East Asian, 3968 Asian unspecified, 2187 South Asian, 3207 European, 1728 Hispanic or Latin American, 4230 African American or Afro-Caribbean, 1277 Oceanian
* [GCST005019](https://www.ebi.ac.uk/gwas/studies/GCST005019), [susceptibility to common cold measurement](https://www.ebi.ac.uk/gwas/efotraits/EFO_0008417), 59,546 European
* [GCST003400](https://www.ebi.ac.uk/gwas/studies/GCST003400), [type II diabetes mellitus](https://www.ebi.ac.uk/gwas/efotraits/EFO_0001360), 41,646 East Asian, 24965 South Asian, 160850 European, 8214 Hispanic or Latin American, 43412 East Asian
* [GCST007847](https://www.ebi.ac.uk/gwas/studies/GCST007847), [type II diabetes mellitus](https://www.ebi.ac.uk/gwas/efotraits/EFO_0001360), 191,764 East Asian
* [GCST010118](https://www.ebi.ac.uk/gwas/studies/GCST010118), [type II diabetes mellitus](https://www.ebi.ac.uk/gwas/efotraits/EFO_0001360), 433,540 East Asian

Download from LD link DB

```bash
Rscript postgwas-exe.r \
  --ldlink down \
  --base   data/20.08_SLC16A11/gwas_biomart_6.tsv \
  --out    ldlink \
  --popul  AMR EAS EUR SAS

# To avoid invalid cross-device link bug
mkdir data/20.08_SLC16A11/ldlink/
mv ldlink/* data/20.08_SLC16A11/ldlink/
rm -r ldlink
```

> ** Run function ldlink_down... ready
> Rsid query = 6.. done
>   Files are moved to target folder:     ldlink
> Job done: 2020-08-02 18:52:59 for 2 min

Filtering LD data by R2 >0.6

```bash
Rscript postgwas-exe.r \
  --ldlink filter \
  --base   data/20.08_SLC16A11/gwas_biomart_6.tsv \
  --ldpath data/20.08_SLC16A11/ldlink \
  --out    data/20.08_SLC16A11 \
  --r2	   0.6
```

> ** Run function ldlink_filter...
> Read download files... 6
>   Read LDlink results           = [1] 3978   12
>     Filtering by "r2 > 0.6": [1] 109  12
>     [Message] No filter criteria for Dprime.
>   Filtered data dimension       = [1] 109   3
>   Excluded no rsid elements     = [1] 0
>
> Basic summary of LDlink results:
>   SNP Tier 1                    = 6
>   SNP Tier 2                    = 26
>   SNP candidates                = 32
>   SNP source annotation table   = [1] 32  2
>
> Add annotations:
>   Calculate LD block index... [1] 3
>
> Search biomart for SNP coordinates:
>   Query SNPs            = [1] 32
>   Hg19 result table     = [1] 32  4
>   Hg38 result table     = [1] 32  4
>   Cytoband annotation... 32.. done
>   Merged table          = [1] 32 11
>
> Write file: data/20.08_SLC16A11/gwas_biomart_32.tsv
> Job done: 2020-08-02 19:17:45 for 8.3 sec



## Preparing input as hg19 BED file

```bash
Rscript postgwas-exe.r \
  --ldlink 	bed \
  --base 	data/20.08_SLC16A11/gwas_biomart_ld_r2_32.tsv \
  --out 	data/20.08_SLC16A11
```

> ** Run function ldlink_bed...
> Read, data/20.08_SLC16A11/gwas_biomart_ld_r2_32.tsv = [1] 32 11
> Write file:     data/20.08_SLC16A11/gwas_hg19_biomart_32.bed
> Write file:     data/20.08_SLC16A11/gwas_hg38_biomart_32.bed
> Job done: 2020-08-02 19:18:07 for 0.2 sec



# 3. Overlapping the functional annotations

## Measuring distance by bedtools

Generating bash script

```bash
Rscript postgwas-exe.r \
  --utils bash \
  --base  data/20.08_SLC16A11/gwas_hg19_biomart_32.bed \
  --ann   data/db_gwas \
  --out   data/20.08_SLC16A11
mv dist_20.08_SLC16A11.sh data/dist_20.08_SLC16A11.sh
```

> ** Run function: utils.r/bash_script... Ready
> Write bash file: 20.08_SLC16A11.sh
> Job done: 2020-08-02 19:41:28 for 0.1 sec

Run bash command

```bash
bash data/dist_20.08_SLC16A11.sh
```



## UCSC gene

Separate ucsc_annot into cds, proximal promoter, whole gene tags

```bash
Rscript postgwas-exe.r \
  --dbfilt    dist \
  --base      data/20.08_SLC16A11/genome_dist/ucsc_annot.tsv \
  --out       data/20.08_SLC16A11/summary \
  --infotype  ucsc
```

> ** Run function: db_filter.r/distance_filt_multi...
> Input file/folder N     = [1] 1
> Input file N    = [1] 1
>
> ** Run function: db_filter.r/distance_filt... Ready
>   File ucsc_annot... nrow= 78.. done
>   Annotations occupied by SNPs  = [1] 17
>   SNPs in annotations           = [1] 19
>
>   UCSC annotations: 3 tags
>     1 cds:      15..    Save at: data/20.08_SLC16A11/summary/snp_ucsc_cds_7.bed
>     2 proximalPromoter: 16.. Save at: data/20.08_SLC16A11/summary/snp_ucsc_proximalPromoter_8.bed
>     3 wholeGene:        28.. Save at: data/20.08_SLC16A11/summary/snp_ucsc_wholeGene_11.bed
> Job done: 2020-08-02 19:58:01 for 0.3 sec



## Roadmap cell type-specific

```bash
Rscript postgwas-exe.r \
  --dbfilt  dist \
  --base    data/20.08_SLC16A11/roadmap_dist \
  --out     data/20.08_SLC16A11/roadmap_over \
  --meta    data/db_gwas/roadmap_meta.tsv
```

> ** Run function: db_filter.r/distance_filt_multi...
> Input file/folder N     = [1] 1
> Input file N    = [1] 127
>   Read metadata file dim        = [1] 127   9
>
> ** Run function: db_filter.r/distance_filt... Ready
>   File roadmap_001_enh... nrow= 32.. done
>   Annotations occupied by SNPs  = [1] 9
>   SNPs in annotations           = [1] 15
>   Write file: data/20.08_SLC16A11/roadmap_over/ESC/snp_roadmap_001_enh_15.bed
> Job process: 0.2 sec
>
> ...
>
> ** Run function: db_filter.r/distance_filt... Ready
>   File roadmap_128_enh... nrow= 32.. done
>   Annotations occupied by SNPs  = [1] 1
>   SNPs in annotations           = [1] 1
>   Write file: data/20.08_SLC16A11/roadmap_over/LUNG_encode/snp_roadmap_128_enh_1.bed
> Job process: 0 sec
>
>
> ** Run function: db_filter.r/distance_filt... Ready
>   File roadmap_129_enh... nrow= 32.. done
>   Annotations occupied by SNPs  = [1] 0
>   SNPs in annotations           = [1] 0
>   [SKIP] SNP N = 0
> Job process: 0 sec
>
> Job done: 2020-08-02 20:01:33 for 2.1 sec



## RegulomeDB

```bash
Rscript postgwas-exe.r \
  --dbfilt  regulome \
  --base    data/20.08_SLC16A11/gwas_hg19_biomart_32.bed \
  --regulm  data/db_gwas/regulome \
  --out     data/20.08_SLC16A11
```

> ** Run function: db_filter.r/regulome_filt... Ready
> Input GWAS SNPs N       = [1] 32
> 2 Regulome data load...
>   Read: data/db_gwas/regulome/dbSNP132.Category1.txt.gz.rds; dim = [1] 39432     5
>   Read: data/db_gwas/regulome/dbSNP132.Category2.txt.gz.rds; dim = [1] 407796      5
>
>   Regulome score >=2b, SNPs             = [1] 430528
>   Functional motifs (1a~2b - 1f only)   = [1] 34705
>
>   Regulome >=2b, GWAS SNPs              = [1] 9
>   GWAS SNPs occupied in
>     functional motifs (1a~2b - 1f only) = [1] 3
>
> Write file: data/20.08_SLC16A11/regulome_9.tsv
> Write file: data/20.08_SLC16A11/summary/snp_regulome2b_9.bed
>
> Job done: 2020-08-02 20:04:34 for 8.1 sec



## GTEx eQTL genes

If the process killed, check the docker system memory size (>8 GB). 

```bash
Rscript postgwas-exe.r \
  --dbfilt gtex_ovl \
  --base   data/20.08_SLC16A11/gwas_hg19_biomart_32.bed \
  --gtex   data/db_gwas/gtex_signif_5e-08_Ensgid.rds \
  --out    data/20.08_SLC16A11
```

> ** Run function: db_filter.r/gtex_overlap... ready
> Input GWAS SNPs N = 32
> Read, gtex_signif_5e-08_Ensgid.rds = [1] 30613850       10
>   Overlapped eQTL-gene pairs = [1] 23
>   eQTLs N = [1] 20
>   Associated eGenes = [1] 2
>
> Write file: data/20.08_SLC16A11/gtex_signif_20.tsv
> Generating BED files at data/20.08_SLC16A11/gtex_eqtl for 2 tissues.. done
>
> Job done: 2020-08-02 20:45:34 for 4.4 min



# 4. Getting Union lists

## ENCODE Tfbs

No overlapped ENCODE Tfbs, look at `20.08_SLC16A11/genome_dist/encode_tfbs.tsv` file.

```bash
Rscript postgwas-exe.r \
  --dbfilt  dist \
  --base    data/20.08_SLC16A11/genome_dist/encode_tfbs.tsv \
  --out     data/20.08_SLC16A11/summary
```

> ** Run function: db_filter.r/distance_filt_multi...
> Input file/folder N     = [1] 1
> Input file N    = [1] 1
>
> ** Run function: db_filter.r/distance_filt... Ready
>   File encode_tfbs... nrow= 32.. Error in `[.data.frame`(rd, , c(8, ncol(rd))) :
>   undefined columns selected
> Calls: db_filter ... lapply -> FUN -> distance_filt -> cbind -> [ -> [.data.frame
> Execution halted



## Roadmap union

```bash
Rscript postgwas-exe.r \
  --dbvenn     summ \
  --base       data/20.08_SLC16A11/roadmap_over \
  --out        data/20.08_SLC16A11/summary \
  --sub_dir    FALSE \
  --uni_save   TRUE
```

> ** Run function: db_venn.r/summ_ann... ready
> 53 Files/folders input.
>   1 3 files in the BRAIN
>   2 1 files in the BRAIN_encode
>   3 2 files in the BREAST
>   4 1 files in the BREAST_encode
>   5 2 files in the Blood_B-cell_CD19p
>   6 1 files in the Blood_HSC_CD34p
>   7 2 files in the Blood_Leukemia_encode
>   8 1 files in the Blood_Lymphoblastoid_encode
>   9 1 files in the Blood_Monocytes_CD14
>   10 1 files in the Blood_Monocytes_CD14_encode
>   11 1 files in the Blood_NK_cell_CD56
>   12 1 files in the Blood_Neutrophils_CD15
>   13 2 files in the Blood_T-cell_CD3
>   14 2 files in the Blood_T-cell_CD8
>   15 1 files in the Blood_Th17_PMA-I_stim_CD4p_CD25m_IL17p
>   16 2 files in the Blood_Th_CD4
>   17 1 files in the Blood_Th_CD4p_CD25m
>   18 1 files in the Blood_Th_PMA-I_stim_CD4p_CD25m_IL17m
>   19 1 files in the Blood_Th_memory_CD4p_CD25m
>   20 1 files in the Blood_Th_naive_CD4p_CD25m
>   21 1 files in the Blood_Tmem_CD4p_CD25int_CD127p
>   22 1 files in the Blood_Treg_CD4p_CD25p
>   23 1 files in the CERVIX_encode
>   24 7 files in the ESC
>   25 8 files in the ESC_DERIVED
>   26 1 files in the FAT
>   27 2 files in the GI_COLON
>   28 2 files in the GI_DUODENUM
>   29 1 files in the GI_ESOPHAGUS
>   30 3 files in the GI_INTESTINE
>   31 3 files in the GI_RECTUM
>   32 2 files in the GI_STOMACH
>   33 3 files in the HEART
>   34 5 files in the IPSC
>   35 1 files in the KIDNEY
>   36 1 files in the LIVER
>   37 1 files in the LIVER_encode
>   38 1 files in the LUNG
>   39 2 files in the LUNG_encode
>   40 2 files in the MUSCLE
>   41 1 files in the MUSCLE_LEG
>   42 2 files in the MUSCLE_encode
>   43 1 files in the OVARY
>   44 1 files in the PANCREAS
>   45 1 files in the PANCREAS_Islets
>   46 2 files in the PLACENTA
>   47 6 files in the SKIN
>   48 2 files in the SKIN_encode
>   49 1 files in the SPLEEN
>   50 1 files in the STROMAL_CONNECTIVE
>   51 2 files in the THYMUS
>   52 1 files in the VASCULAR
>   53 1 files in the VASCULAR_encode
> Total 98 file(s) is/are input.
>
> ** Run function: db_venn.r/venn_bed...
>   Read 98 files
>
> [Message] Can't plot Venn diagram for more than 5 sets.
>
> [Message] Can't plot Euler plot.
>
> ** Back to function: db_venn.r/summ...
>   Returned union list dim = [1]  15 102
>   Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_15.bed
>   [PASS] Nearest gene summary.
>
> Job done: 2020-08-02 20:51:09 for 4.4 sec



## Roadmap sub-groups

```bash
Rscript postgwas-exe.r \
  --dbvenn     summ \
  --base       data/20.08_SLC16A11/roadmap_over \
  --out        data/20.08_SLC16A11/summary \
  --sub_dir    TRUE \
  --uni_save   TRUE
```

> ** Run function: db_venn.r/summ_ann... ready
> 53 Files/folders input.
>   1 sub_dir 1: 3 file(s) in the BRAIN folder
>   2 sub_dir 2: 1 file(s) in the BRAIN_encode folder
>   3 sub_dir 3: 2 file(s) in the BREAST folder
>   4 sub_dir 4: 1 file(s) in the BREAST_encode folder
>   5 sub_dir 5: 2 file(s) in the Blood_B-cell_CD19p folder
>   6 sub_dir 6: 1 file(s) in the Blood_HSC_CD34p folder
>   7 sub_dir 7: 2 file(s) in the Blood_Leukemia_encode folder
>   8 sub_dir 8: 1 file(s) in the Blood_Lymphoblastoid_encode folder
>   9 sub_dir 9: 1 file(s) in the Blood_Monocytes_CD14 folder
>   10 sub_dir 10: 1 file(s) in the Blood_Monocytes_CD14_encode folder
>   11 sub_dir 11: 1 file(s) in the Blood_NK_cell_CD56 folder
>   12 sub_dir 12: 1 file(s) in the Blood_Neutrophils_CD15 folder
>   13 sub_dir 13: 2 file(s) in the Blood_T-cell_CD3 folder
>   14 sub_dir 14: 2 file(s) in the Blood_T-cell_CD8 folder
>   15 sub_dir 15: 1 file(s) in the Blood_Th17_PMA-I_stim_CD4p_CD25m_IL17p folder
>   16 sub_dir 16: 2 file(s) in the Blood_Th_CD4 folder
>   17 sub_dir 17: 1 file(s) in the Blood_Th_CD4p_CD25m folder
>   18 sub_dir 18: 1 file(s) in the Blood_Th_PMA-I_stim_CD4p_CD25m_IL17m folder
>   19 sub_dir 19: 1 file(s) in the Blood_Th_memory_CD4p_CD25m folder
>   20 sub_dir 20: 1 file(s) in the Blood_Th_naive_CD4p_CD25m folder
>   21 sub_dir 21: 1 file(s) in the Blood_Tmem_CD4p_CD25int_CD127p folder
>   22 sub_dir 22: 1 file(s) in the Blood_Treg_CD4p_CD25p folder
>   23 sub_dir 23: 1 file(s) in the CERVIX_encode folder
>   24 sub_dir 24: 7 file(s) in the ESC folder
>   25 sub_dir 25: 8 file(s) in the ESC_DERIVED folder
>   26 sub_dir 26: 1 file(s) in the FAT folder
>   27 sub_dir 27: 2 file(s) in the GI_COLON folder
>   28 sub_dir 28: 2 file(s) in the GI_DUODENUM folder
>   29 sub_dir 29: 1 file(s) in the GI_ESOPHAGUS folder
>   30 sub_dir 30: 3 file(s) in the GI_INTESTINE folder
>   31 sub_dir 31: 3 file(s) in the GI_RECTUM folder
>   32 sub_dir 32: 2 file(s) in the GI_STOMACH folder
>   33 sub_dir 33: 3 file(s) in the HEART folder
>   34 sub_dir 34: 5 file(s) in the IPSC folder
>   35 sub_dir 35: 1 file(s) in the KIDNEY folder
>   36 sub_dir 36: 1 file(s) in the LIVER folder
>   37 sub_dir 37: 1 file(s) in the LIVER_encode folder
>   38 sub_dir 38: 1 file(s) in the LUNG folder
>   39 sub_dir 39: 2 file(s) in the LUNG_encode folder
>   40 sub_dir 40: 2 file(s) in the MUSCLE folder
>   41 sub_dir 41: 1 file(s) in the MUSCLE_LEG folder
>   42 sub_dir 42: 2 file(s) in the MUSCLE_encode folder
>   43 sub_dir 43: 1 file(s) in the OVARY folder
>   44 sub_dir 44: 1 file(s) in the PANCREAS folder
>   45 sub_dir 45: 1 file(s) in the PANCREAS_Islets folder
>   46 sub_dir 46: 2 file(s) in the PLACENTA folder
>   47 sub_dir 47: 6 file(s) in the SKIN folder
>   48 sub_dir 48: 2 file(s) in the SKIN_encode folder
>   49 sub_dir 49: 1 file(s) in the SPLEEN folder
>   50 sub_dir 50: 1 file(s) in the STROMAL_CONNECTIVE folder
>   51 sub_dir 51: 2 file(s) in the THYMUS folder
>   52 sub_dir 52: 1 file(s) in the VASCULAR folder
>   53 sub_dir 53: 1 file(s) in the VASCULAR_encode folder
> Total 53 sub-folder(s) is/are input
> Total 0 file(s) is/are input.
>
> Option sub_dir = TRUE, summary table are not going to be generated.
>   1 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_BRAIN_1.bed
>   2 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_BRAIN_encode_1.bed
>   3 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_BREAST_4.bed
>   4 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_BREAST_encode_2.bed
>   5 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_Blood_B-cell_CD19p_5.bed
>   6 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_Blood_HSC_CD34p_1.bed
>   7 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_Blood_Leukemia_encode_2.bed
>   8 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_Blood_Lymphoblastoid_encode_4.bed
>   9 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_Blood_Monocytes_CD14_3.bed
>   10 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_Blood_Monocytes_CD14_encode_2.bed
>   11 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_Blood_NK_cell_CD56_2.bed
>   12 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_Blood_Neutrophils_CD15_6.bed
>   13 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_Blood_T-cell_CD3_6.bed
>   14 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_Blood_T-cell_CD8_4.bed
>   15 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_Blood_Th17_PMA-I_stim_CD4p_CD25m_IL17p_5.bed
>   16 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_Blood_Th_CD4_5.bed
>   17 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_Blood_Th_CD4p_CD25m_5.bed
>   18 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_Blood_Th_PMA-I_stim_CD4p_CD25m_IL17m_6.bed
>   19 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_Blood_Th_memory_CD4p_CD25m_5.bed
>   20 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_Blood_Th_naive_CD4p_CD25m_5.bed
>   21 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_Blood_Tmem_CD4p_CD25int_CD127p_5.bed
>   22 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_Blood_Treg_CD4p_CD25p_5.bed
>   23 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_CERVIX_encode_4.bed
>   24 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_ESC_15.bed
>   25 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_ESC_DERIVED_5.bed
>   26 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_FAT_1.bed
>   27 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_GI_COLON_1.bed
>   28 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_GI_DUODENUM_2.bed
>   29 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_GI_ESOPHAGUS_3.bed
>   30 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_GI_INTESTINE_3.bed
>   31 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_GI_RECTUM_4.bed
>   32 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_GI_STOMACH_3.bed
>   33 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_HEART_1.bed
>   34 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_IPSC_3.bed
>   35 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_KIDNEY_2.bed
>   36 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_LIVER_3.bed
>   37 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_LIVER_encode_2.bed
>   38 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_LUNG_3.bed
>   39 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_LUNG_encode_1.bed
>   40 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_MUSCLE_2.bed
>   41 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_MUSCLE_LEG_1.bed
>   42 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_MUSCLE_encode_2.bed
>   43 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_OVARY_2.bed
>   44 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_PANCREAS_2.bed
>   45 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_PANCREAS_Islets_1.bed
>   46 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_PLACENTA_3.bed
>   47 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_SKIN_6.bed
>   48 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_SKIN_encode_5.bed
>   49 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_SPLEEN_3.bed
>   50 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_STROMAL_CONNECTIVE_1.bed
>   51 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_THYMUS_2.bed
>   52 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_VASCULAR_1.bed
>   53 Write a BED file: data/20.08_SLC16A11/summary/snp_union_roadmap_over_VASCULAR_encode_2.bed
> Job done: 2020-08-02 20:52:11 for 4.4 sec



## GTEx eQTLs union

```bash
Rscript postgwas-exe.r \
  --dbvenn   summ \
  --base     data/20.08_SLC16A11/gtex_eqtl \
  --out      data/20.08_SLC16A11/gtex_eqtl \
  --sub_dir  FALSE \
  --uni_save TRUE
```

> ** Run function: db_venn.r/summ_ann... ready
> 2 Files/folders input.
>   1 data/20.08_SLC16A11/gtex_eqtl/snp_gtex_Colon_Sigmoid_3.bed
>   2 data/20.08_SLC16A11/gtex_eqtl/snp_gtex_Thyroid_20.bed
> Total 2 file(s) is/are input.
>
> ** Run function: db_venn.r/venn_bed...
>   Read 1: snp_gtex_Colon_Sigmoid_3
>   Read 2: snp_gtex_Thyroid_20
>
> [Message] Can't plot Euler plot.
>
> ** Back to function: db_venn.r/summ...
>   Returned union list dim = [1] 20  6
>   Write a BED file: data/20.08_SLC16A11/gtex_eqtl/snp_union_gtex_eqtl_20.bed
>   [PASS] Nearest gene summary.
>
> Job done: 2020-08-02 20:54:10 for 3.4 sec



# 5. Summary table

```bash
Rscript postgwas-exe.r \
  --dbvenn    summ \
  --base      data/20.08_SLC16A11/summary data/20.08_SLC16A11/gtex_eqtl \
  --out       data/20.08_SLC16A11 \
  --sub_dir   FALSE \
  --uni_save  FALSE \
  --ann_gwas  data/20.08_SLC16A11/gwas_biomart_ld_r2_32.tsv \
  --ann_near  data/20.08_SLC16A11/genome_dist/nearest_gene.tsv \
  --ann_gtex  data/20.08_SLC16A11/gtex_signif_20.tsv
  
  #--ann_encd  data/20.08_SLC16A11/genome_dist/encode_tfbs.tsv \
```

> ** Run function: db_venn.r/summ_ann... ready
> 61 Files/folders input.
>   1 data/20.08_SLC16A11/gtex_eqtl/snp_gtex_Colon_Sigmoid_3.bed
>   2 data/20.08_SLC16A11/gtex_eqtl/snp_gtex_Thyroid_20.bed
>   3 data/20.08_SLC16A11/gtex_eqtl/snp_union_gtex_eqtl_20.bed
>   4 data/20.08_SLC16A11/summary/snp_regulome2b_9.bed
>   5 data/20.08_SLC16A11/summary/snp_ucsc_cds_7.bed
>   6 data/20.08_SLC16A11/summary/snp_ucsc_proximalPromoter_8.bed
>   7 data/20.08_SLC16A11/summary/snp_ucsc_wholeGene_11.bed
>   8 data/20.08_SLC16A11/summary/snp_union_roadmap_over_15.bed
>   9 data/20.08_SLC16A11/summary/snp_union_roadmap_over_BRAIN_1.bed
>   10 data/20.08_SLC16A11/summary/snp_union_roadmap_over_BRAIN_encode_1.bed
>   11 data/20.08_SLC16A11/summary/snp_union_roadmap_over_BREAST_4.bed
>   12 data/20.08_SLC16A11/summary/snp_union_roadmap_over_BREAST_encode_2.bed
>   13 data/20.08_SLC16A11/summary/snp_union_roadmap_over_Blood_B-cell_CD19p_5.bed
>   14 data/20.08_SLC16A11/summary/snp_union_roadmap_over_Blood_HSC_CD34p_1.bed
>   15 data/20.08_SLC16A11/summary/snp_union_roadmap_over_Blood_Leukemia_encode_2.bed
>   16 data/20.08_SLC16A11/summary/snp_union_roadmap_over_Blood_Lymphoblastoid_encode_4.bed
>   17 data/20.08_SLC16A11/summary/snp_union_roadmap_over_Blood_Monocytes_CD14_3.bed
>   18 data/20.08_SLC16A11/summary/snp_union_roadmap_over_Blood_Monocytes_CD14_encode_2.bed
>   19 data/20.08_SLC16A11/summary/snp_union_roadmap_over_Blood_NK_cell_CD56_2.bed
>   20 data/20.08_SLC16A11/summary/snp_union_roadmap_over_Blood_Neutrophils_CD15_6.bed
>   21 data/20.08_SLC16A11/summary/snp_union_roadmap_over_Blood_T-cell_CD3_6.bed
>   22 data/20.08_SLC16A11/summary/snp_union_roadmap_over_Blood_T-cell_CD8_4.bed
>   23 data/20.08_SLC16A11/summary/snp_union_roadmap_over_Blood_Th17_PMA-I_stim_CD4p_CD25m_IL17p_5.bed
>   24 data/20.08_SLC16A11/summary/snp_union_roadmap_over_Blood_Th_CD4_5.bed
>   25 data/20.08_SLC16A11/summary/snp_union_roadmap_over_Blood_Th_CD4p_CD25m_5.bed
>   26 data/20.08_SLC16A11/summary/snp_union_roadmap_over_Blood_Th_PMA-I_stim_CD4p_CD25m_IL17m_6.bed
>   27 data/20.08_SLC16A11/summary/snp_union_roadmap_over_Blood_Th_memory_CD4p_CD25m_5.bed
>   28 data/20.08_SLC16A11/summary/snp_union_roadmap_over_Blood_Th_naive_CD4p_CD25m_5.bed
>   29 data/20.08_SLC16A11/summary/snp_union_roadmap_over_Blood_Tmem_CD4p_CD25int_CD127p_5.bed
>   30 data/20.08_SLC16A11/summary/snp_union_roadmap_over_Blood_Treg_CD4p_CD25p_5.bed
>   31 data/20.08_SLC16A11/summary/snp_union_roadmap_over_CERVIX_encode_4.bed
>   32 data/20.08_SLC16A11/summary/snp_union_roadmap_over_ESC_15.bed
>   33 data/20.08_SLC16A11/summary/snp_union_roadmap_over_ESC_DERIVED_5.bed
>   34 data/20.08_SLC16A11/summary/snp_union_roadmap_over_FAT_1.bed
>   35 data/20.08_SLC16A11/summary/snp_union_roadmap_over_GI_COLON_1.bed
>   36 data/20.08_SLC16A11/summary/snp_union_roadmap_over_GI_DUODENUM_2.bed
>   37 data/20.08_SLC16A11/summary/snp_union_roadmap_over_GI_ESOPHAGUS_3.bed
>   38 data/20.08_SLC16A11/summary/snp_union_roadmap_over_GI_INTESTINE_3.bed
>   39 data/20.08_SLC16A11/summary/snp_union_roadmap_over_GI_RECTUM_4.bed
>   40 data/20.08_SLC16A11/summary/snp_union_roadmap_over_GI_STOMACH_3.bed
>   41 data/20.08_SLC16A11/summary/snp_union_roadmap_over_HEART_1.bed
>   42 data/20.08_SLC16A11/summary/snp_union_roadmap_over_IPSC_3.bed
>   43 data/20.08_SLC16A11/summary/snp_union_roadmap_over_KIDNEY_2.bed
>   44 data/20.08_SLC16A11/summary/snp_union_roadmap_over_LIVER_3.bed
>   45 data/20.08_SLC16A11/summary/snp_union_roadmap_over_LIVER_encode_2.bed
>   46 data/20.08_SLC16A11/summary/snp_union_roadmap_over_LUNG_3.bed
>   47 data/20.08_SLC16A11/summary/snp_union_roadmap_over_LUNG_encode_1.bed
>   48 data/20.08_SLC16A11/summary/snp_union_roadmap_over_MUSCLE_2.bed
>   49 data/20.08_SLC16A11/summary/snp_union_roadmap_over_MUSCLE_LEG_1.bed
>   50 data/20.08_SLC16A11/summary/snp_union_roadmap_over_MUSCLE_encode_2.bed
>   51 data/20.08_SLC16A11/summary/snp_union_roadmap_over_OVARY_2.bed
>   52 data/20.08_SLC16A11/summary/snp_union_roadmap_over_PANCREAS_2.bed
>   53 data/20.08_SLC16A11/summary/snp_union_roadmap_over_PANCREAS_Islets_1.bed
>   54 data/20.08_SLC16A11/summary/snp_union_roadmap_over_PLACENTA_3.bed
>   55 data/20.08_SLC16A11/summary/snp_union_roadmap_over_SKIN_6.bed
>   56 data/20.08_SLC16A11/summary/snp_union_roadmap_over_SKIN_encode_5.bed
>   57 data/20.08_SLC16A11/summary/snp_union_roadmap_over_SPLEEN_3.bed
>   58 data/20.08_SLC16A11/summary/snp_union_roadmap_over_STROMAL_CONNECTIVE_1.bed
>   59 data/20.08_SLC16A11/summary/snp_union_roadmap_over_THYMUS_2.bed
>   60 data/20.08_SLC16A11/summary/snp_union_roadmap_over_VASCULAR_1.bed
>   61 data/20.08_SLC16A11/summary/snp_union_roadmap_over_VASCULAR_encode_2.bed
> Total 61 file(s) is/are input.
>
> ** Run function: db_venn.r/venn_bed...
>   Read 61 files
>
> [Message] Can't plot Venn diagram for more than 5 sets.
>
> [Message] Can't plot Euler plot.
>
> ** Back to function: db_venn.r/summ...
>   Returned union list dim = [1] 32 65
>
>   [PASS] uni_save = FALSE
>
>   GWAS dim = [1] 32 11
>   Merge dim = [1] 32 69
>   Write a CSV file: data/20.08_SLC16A11/summary_gwas.csv
>
>   Nearest gene dim = [1] 32  9
>   Search biomaRt... 3.. Cache found
> 3.. [1] 32  5
>   Merge dim = [1] 32 66
>   Write a CSV file: data/20.08_SLC16A11/summary_nearest.csv
>
>   GTEx dim = [1] 23 10
>   Merge dim = [1] 23 69
>   Write a CSV file: data/20.08_SLC16A11/summary_gtex.csv
>
> Job done: 2020-08-02 21:00:43 for 6.7 sec



## SNP-nexus for AA change annotation

Ref: https://www.snp-nexus.org/v4/results/0c6945b4/

Using the 32 candidate SNPs as query



## LD block region plot

LocusZoom, Ref: http://locuszoom.org/

GWAS Catalog, Ref: https://www.ebi.ac.uk/gwas/variants/rs75418188 <- Picked to draw plot!