# Prolyl hydroxylase ACACB (ACC2) and EGLN3 (PHD3)

## Requirements

Install R in bash

```bash
sudo apt update &&
	sudo apt -y upgrade &&
	sudo apt -y install r-base
```

R libraries
* biomaRt (Bioconductor)
  * Issue for "XML had Non Zero Exit Status" error: https://stackoverflow.com/questions/20671814/non-zero-exit-status-r-3-0-1-xml-and-rcurl
  * `$ sudo apt-get install libcurl4-openssl-dev libxml2-dev`
  * Issue for "openssl had non-zero exit status" error: https://github.com/rocker-org/rocker/issues/124
  * `$ sudo apt install libssl-dev`
  * Issue for "Biobase had non-zero exit status" error: https://stackoverflow.com/questions/56241007/non-zero-exit-status-r-3-6-0-biobase
  * `> Sys.setenv(R_INSTALL_STAGED = FALSE)`
  * Issue for install XML library: https://stackoverflow.com/questions/26042751/cannot-install-package-xml-to-r 
  * `> install.packages("XML", repos = "http://www.omegahat.net/R")`
* limma (Bioconductor)
* dplyr (CRAN)
* data.table (CRAN)
* eulerr (CRAN)
* tools (CRAN)

Install R libraries

```R
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("biomaRt")
BiocManager::install("limma")

install.packages(c('dplyr','data.table','eulerr'))
```

* Install Bedtools: https://bedtools.readthedocs.io/en/latest/content/installation.html



Collaboration for Dr. Marcia C. Haigis

Search results from age-related trait master file v3

| **Traits**                     | **DBs** | **Rsids**                                                    | **Genes** | **P.value** | **Anno**                              |
| ------------------------------ | ------- | ------------------------------------------------------------ | --------- | ----------- | ------------------------------------- |
| ACUTE MYELOID LEUKEMIA         | gwas    | rs1769621; rs10083460; rs1680686; rs1680684; rs1680679;  rs4981207; rs1769628 | EGLN3     | 1.00E-19    | [gwas]_Acute myeloid leukemia         |
| ACUTE MYELOID LEUKEMIA         | gwas    | rs1769621; rs10083460; rs1680686; rs1680684; rs1680679;  rs4981207; rs1769628 | EGLN3     | 5.00E-36    | [gwas]_Acute myeloid leukemia         |
| AGE-RELATED HEARING IMPAIRMENT | gwas    | rs1958581                                                    | EGLN3     | 8.00E-06    | [gwas]_Age-related hearing impairment |
| LATE-ONSET ALZHEIMERS DISEASE  | gwas    | rs35833468                                                   | EGLN3     | 3.00E-06    | [gwas]_Late-onset Alzheimers disease  |
| TRIGLYCERIDE MEASUREMENT       | gwas    | rs149793040                                                  | ACACB     | 3.00E-08    | [gwas]_Triglyceride measurement       |



# 1. Seed SNPs - BED (hg19)

Input Rsid list = `ACACB_EGLN3_risd_10.tsv`

```CMD
Rscript postgwas-exe.r ^
  --dbdown  genebed ^
  --base    Haigis-r0.6/seed_rsid_10.txt ^
  --out     Haigis-r0.6 ^
  --hg      hg19
```

> Read, Haigis-ACACB_EGLN3/ACACB_EGLN3_rsid_10.tsv = [1] 10
> Search biomart for SNP coordinates:
>   Query SNPs            = [1] 10
>   Hg19 result table     = [1] 10  4
>   Cytoband annotation... 10.. done
>   Merged table          = [1] 10  5
>
> Write TSV file: Haigis-ACACB_EGLN3/gwas_biomart_10.tsv
> Write BED file: Haigis-ACACB_EGLN3/gwas_biomart_10.bed
> Job done: 2020-07-21 01:48:21 for 8.5 sec



# 2. Incorporate LD linked SNPs (R2 >0.6)

## Downloading from LDlink DB

All populations:

* AFR African: YRI LWK GWD MSL ESN ASW ACB
* AMR Ad Mixed American: MXL PUR CLM PEL
* EAS East Asian: CHB JPT CHS CDX KHV
* EUR European: CEU TSI FIN GBR IBS
* SAS South Asian: GIH PJL BEB STU ITU

```CMD
Rscript postgwas-exe.r ^
  --ldlink down ^
  --base Haigis-r0.6/seed_rsid_10.txt ^
  --out Haigis-r0.6/ldlink ^
  --popul YRI LWK GWD MSL ESN ASW ACB MXL PUR CLM PEL CHB JPT CHS CDX KHV CEU TSI FIN GBR IBS GIH PJL BEB STU ITU
```

> ** Run function ldlink_down... 10..
> done
>   Files are moved to target folder:     Haigis-r0.6/ldlink
> Job done: 2020-07-22 14:07:56 for 24.1 min



## Filtering the LD link data

```CMD
Rscript postgwas-exe.r ^
  --ldlink 	filter ^
  --base 	Haigis-r0.6/gwas_biomart_10.tsv Haigis-r0.6/ldlink ^
  --out 	Haigis-r0.6 ^
  --r2	 	0.6
```

> ** Run function ldlink_filter...
> Read download files... 10
>   Read LDlink results           = [1] 2622   12
>     Filtering by "r2 > 0.6": [1] 108  12
>     [Msg] No filter criteria for Dprime.
>   Filtered data dimension       = [1] 108   3
>   Excluded no rsid elements     = [1] 1
> Basic summary of LDlink results:
>   SNP Tier 1                    = 10
>   SNP Tier 2                    = 45
>   SNP candidates                = 55
>   SNP source annotation table   = [1] 55  2
> Add annotations:
>   LD block annotation... [1] 6
> Search biomart for SNP coordinates:
>   Query SNPs            = [1] 55
>   Hg19 result table     = [1] 55  4
>   Hg38 result table     = [1] 55  4
>   Cytoband annotation... 55.. done
>
>   Merged table          = [1] 55 11
> Write file: Haigis-r0.6/gwas_biomart_55.tsv
> Job done: 2020-07-22 15:42:07 for 16.6 sec



## Preparing hg19 BED input file

```CMD
Rscript postgwas-exe.r ^
  --ldlink 	bed ^
  --base 	Haigis-r0.6/gwas_biomart_55.tsv ^
  --out 	Haigis-r0.6
```

> ** Run function ldlink_bed...
> [1] 55 11
> Write file:     Haigis-r0.6/gwas_hg19_biomart_55.bed
> Write file:     Haigis-r0.6/gwas_hg38_biomart_55.bed
> Job done: 2020-07-22 15:44:58 for 0.3 sec



# 3. Distances from the annotations

## Generating bash script

```bash
Rscript postgwas-exe.r \
  --utils  bash \
  --base   Haigis-r0.6/gwas_hg19_biomart_55.bed \
  --out    Haigis-r0.6
```

> ** Run function: bedtools.r/bash_script... Ready
> Write bash file: Haigis-r0.6/Haigis-r0.6.sh
> Job done: 2020-07-22 23:33:19 for 0 sec



## Run bash command

Run bash command for bedtools closest

```bash
bash Haigis-r0.6/Haigis-r0.6.sh
```



# 4. Overlapping the annotations

## UCSC gene annotations

```bash
Rscript postgwas-exe.r \
  --dbfilt    dist \
  --base      Haigis-r0.6/genome_dist/ucsc_annot.tsv \
  --out       Haigis-r0.6/summary \
  --infotype  ucsc
```

> Input file/folder N     = [1] 1
>
> ** Run function: db_filter.r/distance_filt_multi...
> Input file N    = [1] 1
>
> ** Run function: db_filter.r/distance_filt... Ready
>   File ucsc_annot... nrow= 67.. done
>   Annotations occupied by SNPs  = [1] 6
>   SNPs in annotations           = [1] 53
>
>   UCSC annotations: 3 tags
>     1 cds:      3..     Save at: Haigis-r0.6/summary/snp_ucsc_cds_1.bed
>     2 proximalPromoter: 1..     Save at: Haigis-r0.6/summary/snp_ucsc_proximalPromoter_1.bed
>     3 wholeGene:        55..    Save at: Haigis-r0.6/summary/snp_ucsc_wholeGene_53.bed
> Job done: 2020-07-22 23:35:07 for 1.6 sec



## Roadmap each cell type

```bash
Rscript postgwas-exe.r \
  --dbfilt  dist \
  --base    Haigis-r0.6/roadmap_dist \
  --out     Haigis-r0.6/roadmap_over \
  --meta    db_gwas/roadmap_meta.tsv
```

> Input file/folder N     = [1] 1
>
> ** Run function: db_filter.r/distance_filt_multi...
> Input file N    = [1] 127
>   Read metadata file dim        = [1] 127   9
>
> ** Run function: db_filter.r/distance_filt... Ready
>   File roadmap_001_enh... nrow= 55.. done
>   Annotations occupied by SNPs  = [1] 5
>   SNPs in annotations           = [1] 29
>   Write file: Haigis-r0.6/roadmap_over/ESC/snp_roadmap_001_enh_29.bed
> Job process: 0.1 sec
>
> ...
>
> ** Run function: db_filter.r/distance_filt... Ready
>   File roadmap_127_enh... nrow= 55.. done
>   Annotations occupied by SNPs  = [1] 1
>   SNPs in annotations           = [1] 1
>   Write file: Haigis-r0.6/roadmap_over/SKIN_encode/snp_roadmap_127_enh_1.bed
> Job process: 0 sec
>
>
> ** Run function: db_filter.r/distance_filt... Ready
>   File roadmap_128_enh... nrow= 55.. done
>   Annotations occupied by SNPs  = [1] 0
>   SNPs in annotations           = [1] 0
>   [SKIP] SNP N = 0
> Job process: 0 sec
>
>
> ** Run function: db_filter.r/distance_filt... Ready
>   File roadmap_129_enh... nrow= 55.. done
>   Annotations occupied by SNPs  = [1] 1
>   SNPs in annotations           = [1] 1
>   Write file: Haigis-r0.6/roadmap_over/BONE_encode/snp_roadmap_129_enh_1.bed
> Job process: 0 sec
>
> Job done: 2020-07-22 23:35:33 for 1.4 sec



## RegulomeDB annotations

```CMD
Rscript postgwas-exe.r \
  --dbfilt  regulome \
  --base    Haigis-r0.6/gwas_hg19_biomart_55.bed \
  --regulm  db_gwas/regulome \
  --out     Haigis-r0.6
```

> ** Run function: db_filter.r/regulome_filt... Ready
> Input GWAS SNPs N       = [1] 55
> 2 Regulome data load...
>      Read: db_gwas/regulome/dbSNP132.Category1.txt.gz.rds; dim = [1] 39432     5
>      Read: db_gwas/regulome/dbSNP132.Category2.txt.gz.rds; dim = [1] 407796      5
>
>      Regulome score >=2b, SNPs             = [1] 430528
>      Functional motifs (1a~2b - 1f only)   = [1] 34705
>
>      Regulome >=2b, GWAS SNPs              = [1] 1
>      GWAS SNPs occupied in
>        functional motifs (1a~2b - 1f only) = [1] 0
>
> Write file: Haigis-r0.6/regulome_1.tsv
> Write file: Haigis-r0.6/summary/snp_regulome2b_1.bed
>
> Job done: 2020-07-22 23:36:26 for 6.4 sec



## GTEx eQTL genes

```bash
Rscript postgwas-exe.r \
  --dbfilt gtex_ovl \
  --base Haigis-r0.6/gwas_hg19_biomart_55.bed \
  --gtex db_gwas/gtex_signif_5e-08_Ensgid.rds \
  --out Haigis-r0.6
```

> ** Run function: db_filter.r/gtex_overlap... Ready
> Input GWAS SNPs N = 55
>     gtex_signif_5e-08.rds, dim = [1] 30613850        9
>     Overlapped eQTL-gene pairs = [1] 0
>     eQTLs N = [1] 0
>     Associated eGenes = [1] 0
>
> [Warning] No overlapped eQTL was found. Please check query SNPs in GTEx homepage.
>
> Job done: 2020-07-22 23:39:39 for 2.6 min



# 4. BED union list

## ENCODE Tfbs

```bash
Rscript postgwas-exe.r \
  --dbfilt  dist \
  --base    Haigis-r0.6/genome_dist/encode_tfbs.tsv \
  --out     Haigis-r0.6/summary
```

> Input file/folder N     = [1] 1
>
> ** Run function: db_filter.r/distance_filt_multi...
> Input file N    = [1] 1
>
> ** Run function: db_filter.r/distance_filt... Ready
>   File encode_tfbs... nrow= 90.. done
>   Annotations occupied by SNPs  = [1] 36
>   SNPs in annotations           = [1] 14
>   Write file: Haigis-r0.6/summary/snp_encode_tfbs_14.bed
> Job done: 2020-07-22 23:40:07 for 0.7 sec



## Roadmap union list

```bash
Rscript postgwas-exe.r \
  --dbvenn     summ \
  --base       Haigis-r0.6/roadmap_over \
  --out        Haigis-r0.6/summary \
  --sub_dir    FALSE \
  --uni_save   TRUE
```

> ** Run function: db_venn.r/summ_ann... ready
> 40 Files/folders input.
>      1 2 files in the Blood_B-cell_CD19p
>      2 4 files in the Blood_HSC_CD34p
>      3 2 files in the Blood_Leukemia_encode
>      4 1 files in the Blood_Lymphoblastoid_encode
>      5 1 files in the Blood_Monocytes_CD14_encode
>      6 1 files in the Blood_Neutrophils_CD15
>      7 1 files in the Blood_T-cell_CD3
>      8 1 files in the Blood_Th_CD4
>      9 1 files in the Blood_Th17_PMA-I_stim_CD4p_CD25m_IL17p
>      10 1 files in the Blood_Tmem_CD4p_CD25int_CD127p
>      11 1 files in the BONE_encode
>      12 11 files in the BRAIN
>      13 1 files in the BRAIN_encode
>      14 2 files in the BREAST
>      15 1 files in the BREAST_encode
>      16 1 files in the CERVIX_encode
>      17 8 files in the ESC
>      18 7 files in the ESC_DERIVED
>      19 3 files in the FAT
>      20 3 files in the GI_COLON
>      21 2 files in the GI_DUODENUM
>      22 1 files in the GI_ESOPHAGUS
>      23 2 files in the GI_INTESTINE
>      24 2 files in the GI_RECTUM
>      25 4 files in the GI_STOMACH
>      26 2 files in the HEART
>      27 5 files in the IPSC
>      28 1 files in the KIDNEY
>      29 1 files in the LIVER
>      30 1 files in the LIVER_encode
>      31 3 files in the LUNG
>      32 1 files in the LUNG_encode
>      33 3 files in the MUSCLE
>      34 1 files in the MUSCLE_encode
>      35 2 files in the PLACENTA
>      36 5 files in the SKIN
>      37 1 files in the SKIN_encode
>      38 2 files in the STROMAL_CONNECTIVE
>      39 2 files in the THYMUS
>      40 1 files in the VASCULAR_encode
>    Total 95 file(s) is/are input.
>    
>    ** Run function: db_venn.r/venn_bed...
>      Read 95 files
>    
>    [Message] Can't plot Venn diagram for more than 5 sets.
>    
>    [Message] Can't plot Euler plot.
>    
>    ** Back to function: db_venn.r/summ...
>      Returned union list dim = [1] 29 99
>      Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_29.bed
>      [PASS] Nearest gene summary.
>    
>    Job done: 2020-07-22 23:54:54 for 2.3 sec

```bash
# Roadmap list by groups
Rscript postgwas-exe.r \
  --dbvenn     summ \
  --base       Haigis-r0.6/roadmap_over \
  --out        Haigis-r0.6/summary \
  --sub_dir    TRUE \
  --uni_save   TRUE
```

> ** Run function: db_venn.r/summ_ann... ready
> 40 Files/folders input.
>     1 sub_dir 1: 2 file(s) in the Blood_B-cell_CD19p folder
>     2 sub_dir 2: 4 file(s) in the Blood_HSC_CD34p folder
>     3 sub_dir 3: 2 file(s) in the Blood_Leukemia_encode folder
>     4 sub_dir 4: 1 file(s) in the Blood_Lymphoblastoid_encode folder
>     5 sub_dir 5: 1 file(s) in the Blood_Monocytes_CD14_encode folder
>     6 sub_dir 6: 1 file(s) in the Blood_Neutrophils_CD15 folder
>     7 sub_dir 7: 1 file(s) in the Blood_T-cell_CD3 folder
>     8 sub_dir 8: 1 file(s) in the Blood_Th_CD4 folder
>     9 sub_dir 9: 1 file(s) in the Blood_Th17_PMA-I_stim_CD4p_CD25m_IL17p folder
>     10 sub_dir 10: 1 file(s) in the Blood_Tmem_CD4p_CD25int_CD127p folder
>     11 sub_dir 11: 1 file(s) in the BONE_encode folder
>     12 sub_dir 12: 11 file(s) in the BRAIN folder
>     13 sub_dir 13: 1 file(s) in the BRAIN_encode folder
>     14 sub_dir 14: 2 file(s) in the BREAST folder
>     15 sub_dir 15: 1 file(s) in the BREAST_encode folder
>     16 sub_dir 16: 1 file(s) in the CERVIX_encode folder
>     17 sub_dir 17: 8 file(s) in the ESC folder
>     18 sub_dir 18: 7 file(s) in the ESC_DERIVED folder
>     19 sub_dir 19: 3 file(s) in the FAT folder
>     20 sub_dir 20: 3 file(s) in the GI_COLON folder
>     21 sub_dir 21: 2 file(s) in the GI_DUODENUM folder
>     22 sub_dir 22: 1 file(s) in the GI_ESOPHAGUS folder
>     23 sub_dir 23: 2 file(s) in the GI_INTESTINE folder
>     24 sub_dir 24: 2 file(s) in the GI_RECTUM folder
>     25 sub_dir 25: 4 file(s) in the GI_STOMACH folder
>     26 sub_dir 26: 2 file(s) in the HEART folder
>     27 sub_dir 27: 5 file(s) in the IPSC folder
>     28 sub_dir 28: 1 file(s) in the KIDNEY folder
>     29 sub_dir 29: 1 file(s) in the LIVER folder
>     30 sub_dir 30: 1 file(s) in the LIVER_encode folder
>     31 sub_dir 31: 3 file(s) in the LUNG folder
>     32 sub_dir 32: 1 file(s) in the LUNG_encode folder
>     33 sub_dir 33: 3 file(s) in the MUSCLE folder
>     34 sub_dir 34: 1 file(s) in the MUSCLE_encode folder
>     35 sub_dir 35: 2 file(s) in the PLACENTA folder
>     36 sub_dir 36: 5 file(s) in the SKIN folder
>     37 sub_dir 37: 1 file(s) in the SKIN_encode folder
>     38 sub_dir 38: 2 file(s) in the STROMAL_CONNECTIVE folder
>     39 sub_dir 39: 2 file(s) in the THYMUS folder
>     40 sub_dir 40: 1 file(s) in the VASCULAR_encode folder
>   Total 40 sub-folder(s) is/are input
>   Total 0 file(s) is/are input.
>   
>   Option sub_dir = TRUE, summary table are not going to be generated.
>     1 Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_Blood_B-cell_CD19p_3.bed
>     2 Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_Blood_HSC_CD34p_5.bed
>     3 Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_Blood_Leukemia_encode_14.bed
>     4 Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_Blood_Lymphoblastoid_encode_3.bed
>     5 Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_Blood_Monocytes_CD14_encode_1.bed
>     6 Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_Blood_Neutrophils_CD15_1.bed
>     7 Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_Blood_T-cell_CD3_1.bed
>     8 Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_Blood_Th_CD4_1.bed
>     9 Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_Blood_Th17_PMA-I_stim_CD4p_CD25m_IL17p_1.bed
>     10 Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_Blood_Tmem_CD4p_CD25int_CD127p_1.bed
>     11 Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_BONE_encode_1.bed
>     12 Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_BRAIN_20.bed
>   13 Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_BRAIN_encode_1.bed
>   14 Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_BREAST_1.bed
>  15 Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_BREAST_encode_1.bed
>   16 Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_CERVIX_encode_2.bed
>     17 Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_ESC_29.bed
>     18 Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_ESC_DERIVED_11.bed
>     19 Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_FAT_1.bed
>     20 Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_GI_COLON_1.bed
>     21 Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_GI_DUODENUM_1.bed
>     22 Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_GI_ESOPHAGUS_1.bed
>     23 Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_GI_INTESTINE_1.bed
>     24 Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_GI_RECTUM_2.bed
>     25 Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_GI_STOMACH_1.bed
>     26 Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_HEART_2.bed
>     27 Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_IPSC_9.bed
>     28 Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_KIDNEY_1.bed
>     29 Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_LIVER_2.bed
>     30 Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_LIVER_encode_1.bed
>     31 Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_LUNG_1.bed
>     32 Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_LUNG_encode_1.bed
>     33 Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_MUSCLE_1.bed
>     34 Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_MUSCLE_encode_1.bed
>     35 Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_PLACENTA_1.bed
>     36 Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_SKIN_1.bed
>     37 Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_SKIN_encode_1.bed
>     38 Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_STROMAL_CONNECTIVE_1.bed
>     39 Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_THYMUS_2.bed
>     40 Write a BED file: Haigis-r0.6/summary/snp_union_roadmap_over_VASCULAR_encode_1.bed
>   Job done: 2020-07-22 23:55:43 for 2.5 sec



## GTEx eQTLs union

```bash
Rscript postgwas-exe.r \
  --dbvenn   summ \
  --base     Haigis-r0.6/gtex_eqtl \
  --out      Haigis-r0.6/gtex_eqtl \
  --sub_dir  FALSE \
  --uni_save TRUE
```

> ** Run function: db_venn.r/summ_ann... ready
> 0 Files/folders input.
>   Job done: 2020-07-22 23:58:29 for 1.9 sec



# 5. Summary tables

````CMD
Rscript postgwas-exe.r \
  --dbvenn    summ \
  --base      Haigis-r0.6/summary \
  --out       Haigis-r0.6 \
  --sub_dir   FALSE \
  --uni_save  FALSE \
  --ann_gwas  Haigis-r0.6/gwas_biomart_55.tsv \
  --ann_encd  Haigis-r0.6/genome_dist/encode_tfbs.tsv \
  --ann_near  Haigis-r0.6/genome_dist/nearest_gene.tsv
````

> ** Run function: db_venn.r/summ_ann... ready
> 46 Files/folders input.
>   1 Haigis-r0.6/summary/snp_encode_tfbs_14.bed
>   2 Haigis-r0.6/summary/snp_regulome2b_1.bed
>   3 Haigis-r0.6/summary/snp_ucsc_cds_1.bed
>   4 Haigis-r0.6/summary/snp_ucsc_proximalPromoter_1.bed
>   5 Haigis-r0.6/summary/snp_ucsc_wholeGene_53.bed
>   6 Haigis-r0.6/summary/snp_union_roadmap_over_29.bed
>   7 Haigis-r0.6/summary/snp_union_roadmap_over_Blood_B-cell_CD19p_3.bed
>   8 Haigis-r0.6/summary/snp_union_roadmap_over_Blood_HSC_CD34p_5.bed
>   9 Haigis-r0.6/summary/snp_union_roadmap_over_Blood_Leukemia_encode_14.bed
>   10 Haigis-r0.6/summary/snp_union_roadmap_over_Blood_Lymphoblastoid_encode_3.bed
>   11 Haigis-r0.6/summary/snp_union_roadmap_over_Blood_Monocytes_CD14_encode_1.bed
>   12 Haigis-r0.6/summary/snp_union_roadmap_over_Blood_Neutrophils_CD15_1.bed
>   13 Haigis-r0.6/summary/snp_union_roadmap_over_Blood_T-cell_CD3_1.bed
>   14 Haigis-r0.6/summary/snp_union_roadmap_over_Blood_Th_CD4_1.bed
>   15 Haigis-r0.6/summary/snp_union_roadmap_over_Blood_Th17_PMA-I_stim_CD4p_CD25m_IL17p_1.bed
>   16 Haigis-r0.6/summary/snp_union_roadmap_over_Blood_Tmem_CD4p_CD25int_CD127p_1.bed
>   17 Haigis-r0.6/summary/snp_union_roadmap_over_BONE_encode_1.bed
>   18 Haigis-r0.6/summary/snp_union_roadmap_over_BRAIN_20.bed
>   19 Haigis-r0.6/summary/snp_union_roadmap_over_BRAIN_encode_1.bed
>   20 Haigis-r0.6/summary/snp_union_roadmap_over_BREAST_1.bed
>   21 Haigis-r0.6/summary/snp_union_roadmap_over_BREAST_encode_1.bed
>   22 Haigis-r0.6/summary/snp_union_roadmap_over_CERVIX_encode_2.bed
>   23 Haigis-r0.6/summary/snp_union_roadmap_over_ESC_29.bed
>   24 Haigis-r0.6/summary/snp_union_roadmap_over_ESC_DERIVED_11.bed
>   25 Haigis-r0.6/summary/snp_union_roadmap_over_FAT_1.bed
>   26 Haigis-r0.6/summary/snp_union_roadmap_over_GI_COLON_1.bed
>   27 Haigis-r0.6/summary/snp_union_roadmap_over_GI_DUODENUM_1.bed
>   28 Haigis-r0.6/summary/snp_union_roadmap_over_GI_ESOPHAGUS_1.bed
>   29 Haigis-r0.6/summary/snp_union_roadmap_over_GI_INTESTINE_1.bed
>   30 Haigis-r0.6/summary/snp_union_roadmap_over_GI_RECTUM_2.bed
>   31 Haigis-r0.6/summary/snp_union_roadmap_over_GI_STOMACH_1.bed
>   32 Haigis-r0.6/summary/snp_union_roadmap_over_HEART_2.bed
>   33 Haigis-r0.6/summary/snp_union_roadmap_over_IPSC_9.bed
>   34 Haigis-r0.6/summary/snp_union_roadmap_over_KIDNEY_1.bed
>   35 Haigis-r0.6/summary/snp_union_roadmap_over_LIVER_2.bed
>   36 Haigis-r0.6/summary/snp_union_roadmap_over_LIVER_encode_1.bed
>   37 Haigis-r0.6/summary/snp_union_roadmap_over_LUNG_1.bed
>   38 Haigis-r0.6/summary/snp_union_roadmap_over_LUNG_encode_1.bed
>   39 Haigis-r0.6/summary/snp_union_roadmap_over_MUSCLE_1.bed
>   40 Haigis-r0.6/summary/snp_union_roadmap_over_MUSCLE_encode_1.bed
>   41 Haigis-r0.6/summary/snp_union_roadmap_over_PLACENTA_1.bed
>   42 Haigis-r0.6/summary/snp_union_roadmap_over_SKIN_1.bed
>   43 Haigis-r0.6/summary/snp_union_roadmap_over_SKIN_encode_1.bed
>   44 Haigis-r0.6/summary/snp_union_roadmap_over_STROMAL_CONNECTIVE_1.bed
>   45 Haigis-r0.6/summary/snp_union_roadmap_over_THYMUS_2.bed
>   46 Haigis-r0.6/summary/snp_union_roadmap_over_VASCULAR_encode_1.bed
> Total 46 file(s) is/are input.
>
> ** Run function: db_venn.r/venn_bed...
>   Read 46 files
>
> [Message] Can't plot Venn diagram for more than 5 sets.
>
> [Message] Can't plot Euler plot.
>
> ** Back to function: db_venn.r/summ...
>   Returned union list dim = [1] 53 50
>
>   [PASS] uni_save = FALSE
>
>   GWAS dim = [1] 55 11
>   Merge dim = [1] 55 54
>   Write a CSV file: Haigis-r0.6/summary_gwas.csv
>
>   ENCODE dim = [1] 90 13
>   Merge dim = [1] 49 49
>   Write a CSV file: Haigis-r0.6/summary_encode.csv
>
>   Nearest gene dim = [1] 55  9
>   Search biomaRt... 4.. Cache found
> 4.. [1] 55  5
>   Merge dim = [1] 55 51
>   Write a CSV file: Haigis-r0.6/summary_nearest.csv
>
> Job done: 2020-07-23 00:04:14 for 5.7 sec



# 6. Search eQTLs from gene query

This step is not developed for command version yet. See details in `Gene query for eQTL.ipynb`.



# * Notes

Ensembl human gene ID to symbol converting tool: http://biotools.fr/human/ensembl_symbol_converter 