# Preterm birth

[ ] Preterm birth or related phenotypes from GWAS/PheWAS/UKB
[ ] How many SNPs/Genes?
[ ] What size of studies (N=?)?



## Brief results



Tissue/cell type-specific high expression

![](20.08-Preterm_birth\Fig.2-SNPsea.png)



## Downloads

Download latest Docker image

```CMD
docker pull kisudsoe/postgwas:latest
```

Download `db_gwas.zip`:  DB files from google drive share point:

* [db_gwas.zip](https://drive.google.com/file/d/1D6CVKi7-83a-g6Gd8X0UhVrYHCcph-wl/view?usp=sharing)



# 1. GWAS Catalog - SNPs

Phenotypes:

1. EFO_0003917-premature_birth.tsv
2. EFO_0006917-spontaneous_preterm_birth.tsv
3. MONDO_0012511-preterm_premature_rupture_of_the_membranes.tsv



## Generating pivot tables

For debugging

```CMD
# For debug 1: Go into the container
docker run -it -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" ^
  kisudsoe/postgwas

# For debug 2: check help message
docker run --rm -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" ^
  kisudsoe/postgwas ^
  Rscript postgwas-exe.r --gwas trait gene study --help
```

Run function

Note: Remove spaces in the file name

```CMD
docker run --rm -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" ^
  kisudsoe/postgwas ^
  Rscript postgwas-exe.r ^
  --gwas trait gene study ^
  --base /data/20.08-Preterm_birth/gwascatalog/EFO_0003917-premature_birth.tsv ^
  --out  /data/20.08-Preterm_birth/gwascatalog
  
docker run --rm -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" ^
  kisudsoe/postgwas ^
  Rscript postgwas-exe.r ^
  --gwas trait gene study ^
  --base /data/20.08-Preterm_birth/gwascatalog/EFO_0006917-spontaneous_preterm_birth.tsv ^
  --out  /data/20.08-Preterm_birth/gwascatalog
  
docker run --rm -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" ^
  kisudsoe/postgwas ^
  Rscript postgwas-exe.r ^
  --gwas trait gene study ^
  --base /data/20.08-Preterm_birth/gwascatalog/MONDO_0012511-preterm_premature_rupture_of_the_membranes.tsv ^
  --out  /data/20.08-Preterm_birth/gwascatalog
```



## Prepare Seed SNPs BED (hg19)

Prepare input file by txt file with 'Rsid' column:

> <...> <Rsid> <...>

Run --dbdown genebed function to get coordinate of SNPs:

```CMD
docker run --rm -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" ^
  kisudsoe/postgwas /bin/sh -c ^
  Rscript postgwas-exe.r ^
  --dbdown genebed ^
  --base   /data/20.08-Preterm_birth/input_snp_149.tsv ^
  --out    /data/20.08-Preterm_birth ^
  --hg     hg19
```

> ** Run function: db_download.r/gene_bed...ready
> Read, ./data/20.08-Preterm_birth/input_snp_149.tsv = [1] 146   1
> Search biomart for SNP coordinates:
>   Query SNPs            = [1] 146
>   Hg19 result table     = [1] 132   4
>   Cytoband annotation... 132.. done
>   Merged table          = [1] 132   5
>
> Write TSV file: ./data/20.08-Preterm_birth/gwas_biomart_132.tsv
> Write BED file: ./data/20.08-Preterm_birth/gwas_biomart_132.bed
> Job done: 2020-08-12 17:09:03 for 47.9 sec



## Incorporate LD linked SNPs (R2 >0.6)

1000 Genome project populations

| Population            | Subpopulations              |
| --------------------- | --------------------------- |
| AFR African           | YRI LWK GWD MSL ESN ASW ACB |
| AMR Ad Mixed American | MXL PUR CLM PEL             |
| EAS East Asian        | CHB JPT CHS CDX KHV         |
| EUR European          | CEU TSI FIN GBR IBS         |
| SAS South Asian       | GIH PJL BEB STU ITU         |

Select populations from 7 studies of 3 traits

1. **Hong X (28598419)**: African American
2. **Jilling T (29538362)**: African American, Hispanic Caucasian, European
3. **Liu X (31477735)**: European, Finnish
4. **Tiensuu H (31194736)**: Finnish
5. **Torgerson DG (30113228)**: European, African American, Hispanic
6. **Zhang G (28877031)**: European
7. **Zhang H (25599974)**: European, African American, Hispanic

Download from LDlink DB by calculating with AMR, EUR populations

```CMD
# To avoid invalid cross-device link error...
docker run --rm -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" ^
  kisudsoe/postgwas /bin/sh -c ^
  "Rscript postgwas-exe.r --ldlink down --base /data/20.08-Preterm_birth/gwas_biomart_132.tsv --out /ldlink --popul AMR EUR && mkdir /data/20.08-Preterm_birth/ldlink && mv /ldlink/* /data/20.08-Preterm_birth/ldlink"
```

> ** Run function ldlink_down... ready
> Rsid query = 132..   error: rs10678727 is not a biallelic variant.,
>   error: rs201450565 is not in 1000G reference panel.,
>   error: rs2115040 is not a biallelic variant.,
>   error: rs35464758 is not a biallelic variant.,
>   error: rs201386833 is not in 1000G reference panel.,
>   error: rs72134357 is not a biallelic variant.,
> done
>   Files are moved to target folder:     /ldlink
> Job done: 2020-08-12 19:23:05 for 25.3 min

Debugging biomart (hg38)

```CMD
docker run --rm -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" ^
  kisudsoe/postgwas Rscript postgwas-exe.r ^
  --ldlink chkbiomart ^
  --out    /data/20.08-Preterm_birth
```

Filtering LD data by R2 >0.6

If you get server connection error, try to change the host address `useast/uswest.ensembl.org`.

```CMD
docker run --rm -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" ^
  kisudsoe/postgwas Rscript postgwas-exe.r ^
  --ldlink filter ^
  --base   /data/20.08-Preterm_birth/gwas_biomart_132.tsv ^
  --ldpath /data/20.08-Preterm_birth/ldlink ^
  --out    /data/20.08-Preterm_birth ^
  --r2     0.6
```

> ** Run function ldlink_filter...
> Read download files... 132
> Error in file(file, "rt") : cannot open the connection
> In addition: Warning message:
> In file(file, "rt") :
>   cannot open file '/data/20.08-Preterm_birth/ldlink/rs10678727.txt': No such file or directory
>   [ERROR] rs10678727
> Error in file(file, "rt") : cannot open the connection
> In addition: Warning message:
> In file(file, "rt") :
>   cannot open file '/data/20.08-Preterm_birth/ldlink/rs201450565.txt': No such file or directory
>   [ERROR] rs201450565
> Error in file(file, "rt") : cannot open the connection
> In addition: Warning message:
> In file(file, "rt") :
>   cannot open file '/data/20.08-Preterm_birth/ldlink/rs2115040.txt': No such file or directory
>   [ERROR] rs2115040
> Error in file(file, "rt") : cannot open the connection
> In addition: Warning message:
> In file(file, "rt") :
>   cannot open file '/data/20.08-Preterm_birth/ldlink/rs35464758.txt': No such file or directory
>   [ERROR] rs35464758
> Error in file(file, "rt") : cannot open the connection
> In addition: Warning message:
> In file(file, "rt") :
>   cannot open file '/data/20.08-Preterm_birth/ldlink/rs201386833.txt': No such file or directory
>   [ERROR] rs201386833
> Error in file(file, "rt") : cannot open the connection
> In addition: Warning message:
> In file(file, "rt") :
>   cannot open file '/data/20.08-Preterm_birth/ldlink/rs72134357.txt': No such file or directory
>   [ERROR] rs72134357
>   Read LDlink results           = [1] 121041     12
>     Filtering by "r2 > 0.6": [1] 2688   12
>     [Message] No filter criteria for Dprime.
>   Filtered data dimension       = [1] 2688    3
>   Excluded no rsid elements     = [1] 20
>
> Basic summary of LDlink results:
>   SNP Tier 1                    = 132
>   SNP Tier 2                    = 2481
>   SNP candidates                = 2613
>   SNP source annotation table   = [1] 2613    2
>
> Add annotations:
>   Calculate LD block index... [1] 122
>
> Search biomart for SNP coordinates:
>   Query SNPs            = [1] 2613
>   Hg19 result table     = [1] 2613    4
>   Hg38 result table     = [1] 2615    4
>   Cytoband annotation... 2628.. done
>   Merged table          = [1] 2628   11
>
> Write file: /data/20.08-Preterm_birth/gwas_biomart_2613.tsv



## Prepare input BED file (hg19)

[Note] Before you run this function, you should fill the `NA` coordinates in hg19 columns as much as you can.

```CMD
docker run --rm -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" ^
  kisudsoe/postgwas Rscript postgwas-exe.r ^
  --ldlink  bed ^
  --base    /data/20.08-Preterm_birth/gwas_biomart_ld_2613.tsv ^
  --out     /data/20.08-Preterm_birth
```

> ** Run function ldlink_bed...
> Read, /data/20.08-Preterm_birth/gwas_biomart_ld_2613.tsv = [1] 2628   11
> Write file:     /data/20.08-Preterm_birth/gwas_hg19_biomart_2616.bed
> Write file:     /data/20.08-Preterm_birth/gwas_hg38_biomart_2615.bed
> Job done: 2020-08-12 20:41:24 for 2.9 sec



# 2. Overlapping the functional annotations



## Measuring distance by bedtools

Generating and run bash script

```CMD
docker run --rm -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" ^
  kisudsoe/postgwas /bin/sh -c ^
  "Rscript postgwas-exe.r --utils bash --base /data/20.08-Preterm_birth/gwas_hg19_biomart_2613.bed --out /data/20.08-Preterm_birth --ann /data/db_gwas && mv dist_20.08-Preterm_birth.sh /data/dist_20.08-Preterm_birth.sh && bash /data/dist_20.08-Preterm_birth.sh"
```

> ** Run function: utils.r/bash_script... Ready
> Write bash file: dist_20.08-Preterm_birth.sh
> Job done: 2020-08-12 20:51:25 for 49.5 sec



## UCSC gene tags separation

Separate ucsc_annot into cds, proximal promoter, whole gene tags

```CMD
docker run --rm -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" ^
  kisudsoe/postgwas Rscript postgwas-exe.r ^
  --dbfilt   dist ^
  --base     /data/20.08-Preterm_birth/genome_dist/ucsc_annot.tsv ^
  --out      /data/20.08-Preterm_birth/summary ^
  --infotype ucsc
```

> ** Run function: db_filter.r/distance_filt... Ready
>   File ucsc_annot... nrow= 11621.. done
>   Annotations occupied by SNPs  = [1] 853
>   SNPs in annotations           = [1] 1733
>
>   UCSC annotations: 3 tags
>     1 cds:      53.. Save at: /data/20.08-Preterm_birth/summary/snp_ucsc_cds_14.bed
>     2 proximalPromoter: 298.. Save at: /data/20.08-Preterm_birth/summary/snp_ucsc_proximalPromoter_188.bed
>     3 wholeGene:        10295.. Save at: /data/20.08-Preterm_birth/summary/snp_ucsc_wholeGene_1708.bed
> Job done: 2020-08-12 21:05:04 for 0.8 sec



## Roadmap merge & cell type

Run enhancer merge

```CMD
docker run --rm -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" ^
  kisudsoe/postgwas Rscript postgwas-exe.r ^
  --dbfilt dist ^
  --base   /data/20.08-Preterm_birth/roadmap_dist/roadmap_enh_merge.tsv ^
  --out    /data/20.08-Preterm_birth/summary
```

> ** Run function: db_filter.r/distance_filt_multi...
> Input file/folder N     = [1] 1
> Input file N    = [1] 1
>
> ** Run function: db_filter.r/distance_filt... Ready
>   File roadmap_enh_merge... nrow= 2613.. done
>   Annotations occupied by SNPs  = [1] 308
>   SNPs in annotations           = [1] 614
>   Write file: /data/20.08-Preterm_birth/summary/snp_roadmap_enh_merge_614.bed
> Job done: 2020-08-14 19:23:27 for 0.5 sec

Group cell types by following meta-information

```CMD
docker run --rm -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" ^
  kisudsoe/postgwas Rscript postgwas-exe.r ^
  --dbfilt dist ^
  --base   /data/20.08-Preterm_birth/roadmap_dist ^
  --out    /data/20.08-Preterm_birth/roadmap_over ^
  --meta   /data/db_gwas/roadmap_meta.tsv
```

> ** Run function: db_filter.r/distance_filt_multi...
> Input file/folder N     = [1] 1
> Input file N    = [1] 128
>   Read metadata file dim        = [1] 127   9
>
> ** Run function: db_filter.r/distance_filt... Ready
>   File roadmap_001_enh... nrow= 2613.. done
>   Annotations occupied by SNPs  = [1] 73
>   SNPs in annotations           = [1] 107
>   Write file: /data/20.08-Preterm_birth/roadmap_over/ESC/snp_roadmap_001_enh_107.bed
> Job process: 0.3 sec
>
> ...
>
> ** Run function: db_filter.r/distance_filt... Ready
>   File roadmap_128_enh... nrow= 2613.. done
>   Annotations occupied by SNPs  = [1] 75
>   SNPs in annotations           = [1] 99
>   Write file: /data/20.08-Preterm_birth/roadmap_over/LUNG_encode/snp_roadmap_128_enh_99.bed
> Job process: 0.1 sec
>
>
> ** Run function: db_filter.r/distance_filt... Ready
>   File roadmap_129_enh... nrow= 2613.. done
>   Annotations occupied by SNPs  = [1] 86
>   SNPs in annotations           = [1] 116
>   Write file: /data/20.08-Preterm_birth/roadmap_over/BONE_encode/snp_roadmap_129_enh_116.bed
> Job process: 0.1 sec
>
> [BREAK] roadmap_enh_merge.tsv is not existing in meta file.
> Job done: 2020-08-15 04:44:17 for 12.5 sec



## RegulomeDB

```CMD
docker run --rm -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" ^
  kisudsoe/postgwas Rscript postgwas-exe.r ^
  --dbfilt  regulome ^
  --base    /data/20.08-Preterm_birth/gwas_hg19_biomart_2613.bed ^
  --out     /data/20.08-Preterm_birth ^
  --regulm  /data/db_gwas/regulome
```

> ** Run function: db_filter.r/regulome_filt... Ready
> Input GWAS SNPs N       = [1] 2613
> 2 Regulome data load...
>   Read: /data/db_gwas/regulome/dbSNP132.Category1.txt.gz.rds; dim = [1] 39432     5
>   Read: /data/db_gwas/regulome/dbSNP132.Category2.txt.gz.rds; dim = [1] 407796      5
>
>   Regulome score >=2b, SNPs             = [1] 430528
>   Functional motifs (1a~2b - 1f only)   = [1] 34705
>
>   Regulome >=2b, GWAS SNPs              = [1] 34
>   GWAS SNPs occupied in
>     functional motifs (1a~2b - 1f only) = [1] 10
>
> Write file: /data/20.08-Preterm_birth/regulome_34.tsv
> Write file: /data/20.08-Preterm_birth/summary/snp_regulome2b_34.bed



## lncRNASNP2

```CMD
docker run --rm -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" ^
  kisudsoe/postgwas Rscript postgwas-exe.r ^
  --dbfilt  lnc_ovl ^
  --base    /data/20.08-Preterm_birth/gwas_hg19_biomart_2613.bed ^
  --out     /data/20.08-Preterm_birth ^
  --lncrna  /data/db_gwas/lncrna
```

> ** Run function: db_filter.r/lncrna_overlap... Ready
> Input GWAS SNPs N = 2613
> 3 lncRNASNP2 data load...
>   Read: /data/db_gwas/lncrna/lncRNASNP2_snplist.txt.rds, = [1] 10205295        3
>   Read: /data/db_gwas/lncrna/lncrnas.txt.rds, = [1] 141271      4
>   Read: /data/db_gwas/lncrna/lncrna-diseases_experiment.txt.rds, = [1] 753   3
>
> Summary =
>   lncRNA SNPs
> 1     84  128
>
>   Write file: /data/20.08-Preterm_birth/summary/snp_lncrnasnp_128.bed
>   Write file: /data/20.08-Preterm_birth/lncrnasnp_128.tsv
> Job done: 2020-08-14 18:38:36 for 38.1 sec



## GTEx eQTL genes

If the process killed during loading GTEx dataset, check the docker system memory size (>10 GB). 

```CMD
docker run --rm ^
  -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" ^
  kisudsoe/postgwas Rscript postgwas-exe.r ^
  --dbfilt  gtex_ovl ^
  --base    /data/20.08-Preterm_birth/gwas_hg19_biomart_2613.bed ^
  --out     /data/20.08-Preterm_birth ^
  --gtex    /data/db_gwas/gtex_signif_5e-08_Ensgid_dt.rds
```

> ** Run function: db_filter.r/gtex_overlap...   Your memory size = [1] Inf
> ready
> Input GWAS SNPs N = 2613
> Read, gtex_signif_5e-08_Ensgid_dt.rds = [1] 30613850        7
>   Overlapped eQTL-gene pairs = [1] 12796
>   eQTLs N = [1] 815
>   Associated eGenes = [1] 0
>
> Write file: /data/20.08-Preterm_birth/gtex_signif_815.tsv
> Generating BED files at "/data/20.08-Preterm_birth/gtex_eqtl" for 38 tissues.. done
>
> Job done: 2020-08-13 01:16:37 for 4.8 min



# 3. Union list to summary



## ENCODE Tfbs

```CMD
docker run --rm ^
  -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" ^
  kisudsoe/postgwas Rscript postgwas-exe.r ^
  --dbfilt dist ^
  --base   /data/20.08-Preterm_birth/genome_dist/encode_tfbs.tsv ^
  --out    /data/20.08-Preterm_birth/summary
```

> ** Run function: db_filter.r/distance_filt_multi...
> Input file/folder N     = [1] 1
> Input file N    = [1] 1
>
> ** Run function: db_filter.r/distance_filt... Ready
>   File encode_tfbs... nrow= 3357.. done
>   Annotations occupied by SNPs  = [1] 899
>   SNPs in annotations           = [1] 280
>   Write file: /data/20.08-Preterm_birth/summary/snp_encode_tfbs_280.bed
> Job done: 2020-08-13 01:18:44 for 0.4 sec



## Roadmap Enhancers

Roadmap union

```CMD
docker run --rm ^
  -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" ^
  kisudsoe/postgwas Rscript postgwas-exe.r ^
  --dbvenn   summ ^
  --base     /data/20.08-Preterm_birth/roadmap_over ^
  --out      /data/20.08-Preterm_birth/summary ^
  --subdir   FALSE ^
  --uni_save TRUE
```

> ** Run function: db_venn.r/summ_ann... ready
> 56 Files/folders input.
>   1 1 files in the ADRENAL
>   2 1 files in the BONE_encode
>   3 12 files in the BRAIN
>   4 1 files in the BRAIN_encode
>   5 2 files in the BREAST
>   6 1 files in the BREAST_encode
>   7 2 files in the Blood_B-cell_CD19p
>   8 4 files in the Blood_HSC_CD34p
>   9 2 files in the Blood_Leukemia_encode
>   10 1 files in the Blood_Lymphoblastoid_encode
>   11 1 files in the Blood_Monocytes_CD14
>   12 1 files in the Blood_Monocytes_CD14_encode
>   13 1 files in the Blood_Mononuclear_cell
>   14 1 files in the Blood_NK_cell_CD56
>   15 1 files in the Blood_Neutrophils_CD15
>   16 2 files in the Blood_T-cell_CD3
>   17 2 files in the Blood_T-cell_CD8
>   18 1 files in the Blood_Th17_PMA-I_stim_CD4p_CD25m_IL17p
>   19 2 files in the Blood_Th_CD4
>   20 1 files in the Blood_Th_CD4p_CD25m
>   21 1 files in the Blood_Th_PMA-I_stim_CD4p_CD25m_IL17m
>   22 1 files in the Blood_Th_memory_CD4p_CD25m
>   23 1 files in the Blood_Th_naive_CD4p_CD25m
>   24 1 files in the Blood_Tmem_CD4p_CD25int_CD127p
>   25 1 files in the Blood_Treg_CD4p_CD25p
>   26 1 files in the CERVIX_encode
>   27 8 files in the ESC
>   28 9 files in the ESC_DERIVED
>   29 3 files in the FAT
>   30 3 files in the GI_COLON
>   31 2 files in the GI_DUODENUM
>   32 1 files in the GI_ESOPHAGUS
>   33 3 files in the GI_INTESTINE
>   34 3 files in the GI_RECTUM
>   35 4 files in the GI_STOMACH
>   36 4 files in the HEART
>   37 5 files in the IPSC
>   38 1 files in the KIDNEY
>   39 1 files in the LIVER
>   40 1 files in the LIVER_encode
>   41 3 files in the LUNG
>   42 2 files in the LUNG_encode
>   43 5 files in the MUSCLE
>   44 1 files in the MUSCLE_LEG
>   45 2 files in the MUSCLE_encode
>   46 1 files in the OVARY
>   47 1 files in the PANCREAS
>   48 1 files in the PANCREAS_Islets
>   49 2 files in the PLACENTA
>   50 6 files in the SKIN
>   51 2 files in the SKIN_encode
>   52 1 files in the SPLEEN
>   53 2 files in the STROMAL_CONNECTIVE
>   54 2 files in the THYMUS
>   55 1 files in the VASCULAR
>   56 1 files in the VASCULAR_encode
> Total 127 file(s) is/are input.
>
> ** Run function: db_venn.r/venn_bed...
>   Read 127 files
>
> [Message] Can't plot Venn diagram for more than 5 sets.
>
> [Message] Can't plot Euler plot.
>
> ** Back to function: db_venn.r/summ...
>   Returned union list dim = [1] 614 131
>   Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_614.bed
>   [PASS] Nearest gene summary.

Roadmap sub-groups

```CMD
docker run --rm ^
  -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" ^
  kisudsoe/postgwas Rscript postgwas-exe.r ^
  --dbvenn   summ ^
  --base     /data/20.08-Preterm_birth/roadmap_over ^
  --out      /data/20.08-Preterm_birth/summary ^
  --sub_dir  TRUE ^
  --uni_save TRUE
```

> ** Run function: db_venn.r/summ_ann... ready
> 56 Files/folders input.
>   1 sub_dir 1: 1 file(s) in the ADRENAL folder
>   2 sub_dir 2: 1 file(s) in the BONE_encode folder
>   3 sub_dir 3: 12 file(s) in the BRAIN folder
>   4 sub_dir 4: 1 file(s) in the BRAIN_encode folder
>   5 sub_dir 5: 2 file(s) in the BREAST folder
>   6 sub_dir 6: 1 file(s) in the BREAST_encode folder
>   7 sub_dir 7: 2 file(s) in the Blood_B-cell_CD19p folder
>   8 sub_dir 8: 4 file(s) in the Blood_HSC_CD34p folder
>   9 sub_dir 9: 2 file(s) in the Blood_Leukemia_encode folder
>   10 sub_dir 10: 1 file(s) in the Blood_Lymphoblastoid_encode folder
>   11 sub_dir 11: 1 file(s) in the Blood_Monocytes_CD14 folder
>   12 sub_dir 12: 1 file(s) in the Blood_Monocytes_CD14_encode folder
>   13 sub_dir 13: 1 file(s) in the Blood_Mononuclear_cell folder
>   14 sub_dir 14: 1 file(s) in the Blood_NK_cell_CD56 folder
>   15 sub_dir 15: 1 file(s) in the Blood_Neutrophils_CD15 folder
>   16 sub_dir 16: 2 file(s) in the Blood_T-cell_CD3 folder
>   17 sub_dir 17: 2 file(s) in the Blood_T-cell_CD8 folder
>   18 sub_dir 18: 1 file(s) in the Blood_Th17_PMA-I_stim_CD4p_CD25m_IL17p folder
>   19 sub_dir 19: 2 file(s) in the Blood_Th_CD4 folder
>   20 sub_dir 20: 1 file(s) in the Blood_Th_CD4p_CD25m folder
>   21 sub_dir 21: 1 file(s) in the Blood_Th_PMA-I_stim_CD4p_CD25m_IL17m folder
>   22 sub_dir 22: 1 file(s) in the Blood_Th_memory_CD4p_CD25m folder
>   23 sub_dir 23: 1 file(s) in the Blood_Th_naive_CD4p_CD25m folder
>   24 sub_dir 24: 1 file(s) in the Blood_Tmem_CD4p_CD25int_CD127p folder
>   25 sub_dir 25: 1 file(s) in the Blood_Treg_CD4p_CD25p folder
>   26 sub_dir 26: 1 file(s) in the CERVIX_encode folder
>   27 sub_dir 27: 8 file(s) in the ESC folder
>   28 sub_dir 28: 9 file(s) in the ESC_DERIVED folder
>   29 sub_dir 29: 3 file(s) in the FAT folder
>   30 sub_dir 30: 3 file(s) in the GI_COLON folder
>   31 sub_dir 31: 2 file(s) in the GI_DUODENUM folder
>   32 sub_dir 32: 1 file(s) in the GI_ESOPHAGUS folder
>   33 sub_dir 33: 3 file(s) in the GI_INTESTINE folder
>   34 sub_dir 34: 3 file(s) in the GI_RECTUM folder
>   35 sub_dir 35: 4 file(s) in the GI_STOMACH folder
>   36 sub_dir 36: 4 file(s) in the HEART folder
>   37 sub_dir 37: 5 file(s) in the IPSC folder
>   38 sub_dir 38: 1 file(s) in the KIDNEY folder
>   39 sub_dir 39: 1 file(s) in the LIVER folder
>   40 sub_dir 40: 1 file(s) in the LIVER_encode folder
>   41 sub_dir 41: 3 file(s) in the LUNG folder
>   42 sub_dir 42: 2 file(s) in the LUNG_encode folder
>   43 sub_dir 43: 5 file(s) in the MUSCLE folder
>   44 sub_dir 44: 1 file(s) in the MUSCLE_LEG folder
>   45 sub_dir 45: 2 file(s) in the MUSCLE_encode folder
>   46 sub_dir 46: 1 file(s) in the OVARY folder
>   47 sub_dir 47: 1 file(s) in the PANCREAS folder
>   48 sub_dir 48: 1 file(s) in the PANCREAS_Islets folder
>   49 sub_dir 49: 2 file(s) in the PLACENTA folder
>   50 sub_dir 50: 6 file(s) in the SKIN folder
>   51 sub_dir 51: 2 file(s) in the SKIN_encode folder
>   52 sub_dir 52: 1 file(s) in the SPLEEN folder
>   53 sub_dir 53: 2 file(s) in the STROMAL_CONNECTIVE folder
>   54 sub_dir 54: 2 file(s) in the THYMUS folder
>   55 sub_dir 55: 1 file(s) in the VASCULAR folder
>   56 sub_dir 56: 1 file(s) in the VASCULAR_encode folder
> Total 56 sub-folder(s) is/are input
> Total 0 file(s) is/are input.
>
> Option sub_dir = TRUE, summary table are not going to be generated.
>   1 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_ADRENAL_82.bed
>   2 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_BONE_encode_116.bed
>   3 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_BRAIN_188.bed
>   4 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_BRAIN_encode_105.bed
>   5 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_BREAST_112.bed
>   6 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_BREAST_encode_76.bed
>   7 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_Blood_B-cell_CD19p_108.bed
>   8 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_Blood_HSC_CD34p_133.bed
>   9 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_Blood_Leukemia_encode_83.bed
>   10 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_Blood_Lymphoblastoid_encode_59.bed
>   11 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_Blood_Monocytes_CD14_71.bed
>   12 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_Blood_Monocytes_CD14_encode_82.bed
>   13 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_Blood_Mononuclear_cell_34.bed
>   14 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_Blood_NK_cell_CD56_41.bed
>   15 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_Blood_Neutrophils_CD15_100.bed
>   16 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_Blood_T-cell_CD3_69.bed
>   17 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_Blood_T-cell_CD8_29.bed
>   18 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_Blood_Th17_PMA-I_stim_CD4p_CD25m_IL17p_22.bed
>   19 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_Blood_Th_CD4_24.bed
>   20 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_Blood_Th_CD4p_CD25m_20.bed
>   21 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_Blood_Th_PMA-I_stim_CD4p_CD25m_IL17m_21.bed
>   22 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_Blood_Th_memory_CD4p_CD25m_17.bed
>   23 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_Blood_Th_naive_CD4p_CD25m_17.bed
>   24 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_Blood_Tmem_CD4p_CD25int_CD127p_24.bed
>   25 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_Blood_Treg_CD4p_CD25p_21.bed
>   26 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_CERVIX_encode_75.bed
>   27 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_ESC_614.bed
>   28 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_ESC_DERIVED_233.bed
>   29 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_FAT_197.bed
>   30 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_GI_COLON_116.bed
>   31 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_GI_DUODENUM_110.bed
>   32 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_GI_ESOPHAGUS_39.bed
>   33 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_GI_INTESTINE_102.bed
>   34 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_GI_RECTUM_96.bed
>   35 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_GI_STOMACH_139.bed
>   36 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_HEART_109.bed
>   37 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_IPSC_175.bed
>   38 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_KIDNEY_88.bed
>   39 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_LIVER_78.bed
>   40 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_LIVER_encode_70.bed
>   41 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_LUNG_140.bed
>   42 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_LUNG_encode_130.bed
>   43 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_MUSCLE_207.bed
>   44 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_MUSCLE_LEG_82.bed
>   45 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_MUSCLE_encode_112.bed
>   46 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_OVARY_60.bed
>   47 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_PANCREAS_46.bed
>   48 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_PANCREAS_Islets_61.bed
>   49 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_PLACENTA_117.bed
>   50 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_SKIN_143.bed
>   51 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_SKIN_encode_140.bed
>   52 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_SPLEEN_60.bed
>   53 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_STROMAL_CONNECTIVE_140.bed
>   54 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_THYMUS_50.bed
>   55 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_VASCULAR_55.bed
>   56 Write a BED file: /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_VASCULAR_encode_89.bed
> Job done: 2020-08-13 01:58:13 for 5.4 sec



## GTEx eQTLs union

```CMD
docker run --rm ^
  -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" ^
  kisudsoe/postgwas Rscript postgwas-exe.r ^
  --dbvenn   summ ^
  --base     /data/20.08-Preterm_birth/gtex_eqtl ^
  --out      /data/20.08-Preterm_birth/gtex_eqtl ^
  --subdir   FALSE ^
  --uni_save TRUE
```

> ** Run function: db_venn.r/summ_ann... ready
> 38 Files/folders input.
>   1 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Adipose_Subcutaneous_352.bed
>   2 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Adipose_Visceral_Omentum_313.bed
>   3 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Adrenal_Gland_31.bed
>   4 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Artery_Aorta_308.bed
>   5 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Artery_Coronary_30.bed
>   6 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Artery_Tibial_379.bed
>   7 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Brain_Anterior_cingulate_cortex_BA24_1.bed
>   8 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Brain_Caudate_basal_ganglia_25.bed
>   9 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Brain_Cerebellar_Hemisphere_22.bed
>   10 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Brain_Cerebellum_63.bed
>   11 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Brain_Cortex_259.bed
>   12 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Brain_Frontal_Cortex_BA9_25.bed
>   13 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Breast_Mammary_Tissue_305.bed
>   14 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Cells_Cultured_fibroblasts_467.bed
>   15 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Colon_Sigmoid_320.bed
>   16 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Colon_Transverse_301.bed
>   17 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Esophagus_Gastroesophageal_Junction_335.bed
>   18 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Esophagus_Mucosa_302.bed
>   19 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Esophagus_Muscularis_345.bed
>   20 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Heart_Atrial_Appendage_285.bed
>   21 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Heart_Left_Ventricle_39.bed
>   22 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Liver_9.bed
>   23 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Lung_308.bed
>   24 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Muscle_Skeletal_168.bed
>   25 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Nerve_Tibial_390.bed
>   26 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Ovary_98.bed
>   27 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Pancreas_34.bed
>   28 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Pituitary_17.bed
>   29 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Prostate_265.bed
>   30 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Skin_Not_Sun_Exposed_Suprapubic_302.bed
>   31 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Skin_Sun_Exposed_Lower_leg_327.bed
>   32 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Small_Intestine_Terminal_Ileum_40.bed
>   33 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Spleen_260.bed
>   34 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Stomach_48.bed
>   35 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Testis_516.bed
>   36 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Thyroid_293.bed
>   37 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Uterus_35.bed
>   38 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Whole_Blood_19.bed
> Total 38 file(s) is/are input.
>
> ** Run function: db_venn.r/venn_bed...
>   Read 38 files
>
> [Message] Can't plot Venn diagram for more than 5 sets.
>
> [Message] Can't plot Euler plot.
>
> ** Back to function: db_venn.r/summ...
>   Returned union list dim = [1] 815  42
>   Write a BED file: /data/20.08-Preterm_birth/gtex_eqtl/snp_union_gtex_eqtl_815.bed
>   [PASS] Nearest gene summary.
>
> Job done: 2020-08-13 01:29:51 for 3.5 sec



# 4. Summary table and run external algorithms

Fix source code in local and Update to Docker image

```CMD
docker run -it -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" ^
  kisudsoe/postgwas:latest /bin/bash
```

```bash
cp -r /data/src /data/postgwas-exe.r /
```

```CMD
docker ps
```

```CMD
docker stop b6a687586b91
docker commit -a "jjy" b6a687586b91 kisudsoe/postgwas:latest
```

Run in image

```CMD
docker run --rm ^
  -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" ^
  kisudsoe/postgwas Rscript postgwas-exe.r ^
  --dbvenn   summ ^
  --base     /data/20.08-Preterm_birth/summary /data/20.08-Preterm_birth/gtex_eqtl ^
  --out      /data/20.08-Preterm_birth ^
  --sub_dir  FALSE ^
  --uni_save FALSE ^
  --ann_gwas /data/20.08-Preterm_birth/gwas_biomart_ld_2613.tsv ^
  --ann_near /data/20.08-Preterm_birth/genome_dist/nearest_gene.tsv ^
  --ann_encd /data/20.08-Preterm_birth/genome_dist/encode_tfbs.tsv ^
  --ann_gtex /data/20.08-Preterm_birth/gtex_signif_815.tsv
```

> ** Run function: db_venn.r/summ_ann... ready
> 104 Files/folders input.
>   1 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Adipose_Subcutaneous_352.bed
>   2 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Adipose_Visceral_Omentum_313.bed
>   3 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Adrenal_Gland_31.bed
>   4 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Artery_Aorta_308.bed
>   5 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Artery_Coronary_30.bed
>   6 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Artery_Tibial_379.bed
>   7 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Brain_Anterior_cingulate_cortex_BA24_1.bed
>   8 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Brain_Caudate_basal_ganglia_25.bed
>   9 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Brain_Cerebellar_Hemisphere_22.bed
>   10 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Brain_Cerebellum_63.bed
>   11 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Brain_Cortex_259.bed
>   12 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Brain_Frontal_Cortex_BA9_25.bed
>   13 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Breast_Mammary_Tissue_305.bed
>   14 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Cells_Cultured_fibroblasts_467.bed
>   15 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Colon_Sigmoid_320.bed
>   16 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Colon_Transverse_301.bed
>   17 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Esophagus_Gastroesophageal_Junction_335.bed
>   18 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Esophagus_Mucosa_302.bed
>   19 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Esophagus_Muscularis_345.bed
>   20 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Heart_Atrial_Appendage_285.bed
>   21 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Heart_Left_Ventricle_39.bed
>   22 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Liver_9.bed
>   23 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Lung_308.bed
>   24 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Muscle_Skeletal_168.bed
>   25 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Nerve_Tibial_390.bed
>   26 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Ovary_98.bed
>   27 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Pancreas_34.bed
>   28 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Pituitary_17.bed
>   29 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Prostate_265.bed
>   30 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Skin_Not_Sun_Exposed_Suprapubic_302.bed
>   31 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Skin_Sun_Exposed_Lower_leg_327.bed
>   32 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Small_Intestine_Terminal_Ileum_40.bed
>   33 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Spleen_260.bed
>   34 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Stomach_48.bed
>   35 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Testis_516.bed
>   36 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Thyroid_293.bed
>   37 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Uterus_35.bed
>   38 /data/20.08-Preterm_birth/gtex_eqtl/snp_gtex_Whole_Blood_19.bed
>   39 /data/20.08-Preterm_birth/gtex_eqtl/snp_union_gtex_eqtl_815.bed
>   40 /data/20.08-Preterm_birth/summary/snp_encode_tfbs_280.bed
>   41 /data/20.08-Preterm_birth/summary/snp_lncrnasnp_128.bed
>   42 /data/20.08-Preterm_birth/summary/snp_regulome2b_34.bed
>   43 /data/20.08-Preterm_birth/summary/snp_roadmap_enh_merge_614.bed
>   44 /data/20.08-Preterm_birth/summary/snp_ucsc_cds_14.bed
>   45 /data/20.08-Preterm_birth/summary/snp_ucsc_proximalPromoter_188.bed
>   46 /data/20.08-Preterm_birth/summary/snp_ucsc_wholeGene_1708.bed
>   47 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_614.bed
>   48 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_ADRENAL_82.bed
>   49 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_BONE_encode_116.bed
>   50 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_BRAIN_188.bed
>   51 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_BRAIN_encode_105.bed
>   52 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_BREAST_112.bed
>   53 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_BREAST_encode_76.bed
>   54 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_Blood_B-cell_CD19p_108.bed
>   55 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_Blood_HSC_CD34p_133.bed
>   56 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_Blood_Leukemia_encode_83.bed
>   57 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_Blood_Lymphoblastoid_encode_59.bed
>   58 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_Blood_Monocytes_CD14_71.bed
>   59 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_Blood_Monocytes_CD14_encode_82.bed
>   60 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_Blood_Mononuclear_cell_34.bed
>   61 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_Blood_NK_cell_CD56_41.bed
>   62 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_Blood_Neutrophils_CD15_100.bed
>   63 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_Blood_T-cell_CD3_69.bed
>   64 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_Blood_T-cell_CD8_29.bed
>   65 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_Blood_Th17_PMA-I_stim_CD4p_CD25m_IL17p_22.bed
>   66 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_Blood_Th_CD4_24.bed
>   67 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_Blood_Th_CD4p_CD25m_20.bed
>   68 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_Blood_Th_PMA-I_stim_CD4p_CD25m_IL17m_21.bed
>   69 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_Blood_Th_memory_CD4p_CD25m_17.bed
>   70 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_Blood_Th_naive_CD4p_CD25m_17.bed
>   71 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_Blood_Tmem_CD4p_CD25int_CD127p_24.bed
>   72 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_Blood_Treg_CD4p_CD25p_21.bed
>   73 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_CERVIX_encode_75.bed
>   74 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_ESC_179.bed
>   75 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_ESC_614.bed
>   76 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_ESC_DERIVED_233.bed
>   77 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_FAT_197.bed
>   78 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_GI_COLON_116.bed
>   79 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_GI_DUODENUM_110.bed
>   80 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_GI_ESOPHAGUS_39.bed
>   81 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_GI_INTESTINE_102.bed
>   82 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_GI_RECTUM_96.bed
>   83 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_GI_STOMACH_139.bed
>   84 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_HEART_109.bed
>   85 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_IPSC_175.bed
>   86 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_KIDNEY_88.bed
>   87 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_LIVER_78.bed
>   88 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_LIVER_encode_70.bed
>   89 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_LUNG_140.bed
>   90 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_LUNG_encode_130.bed
>   91 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_MUSCLE_207.bed
>   92 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_MUSCLE_LEG_82.bed
>   93 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_MUSCLE_encode_112.bed
>   94 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_OVARY_60.bed
>   95 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_PANCREAS_46.bed
>   96 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_PANCREAS_Islets_61.bed
>   97 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_PLACENTA_117.bed
>   98 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_SKIN_143.bed
>   99 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_SKIN_encode_140.bed
>   100 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_SPLEEN_60.bed
>   101 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_STROMAL_CONNECTIVE_140.bed
>   102 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_THYMUS_50.bed
>   103 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_VASCULAR_55.bed
>   104 /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_VASCULAR_encode_89.bed
> Total 104 file(s) is/are input.
>
> ** Run function: db_venn.r/venn_bed...
>   Read 104 files
>
>
> [Message] Can't plot Venn diagram for more than 5 sets.
>
> [Message] Can't plot Euler plot.
> ** Back to function: db_venn.r/summ...
>   Returned union list dim = [1] 2055  108
>   [PASS] uni_save = FALSE
>
>   ENCODE dim = [1] 3357   13
>   Merge dim = [1] 1008  107
>   Extract snp-tfbs pair = [1] 280   2
>   Write a CSV file: /data/20.08-Preterm_birth/summary_encode.csv
>
>   Nearest gene dim = [1] 2807    9
>   Search biomaRt... 207.. parsing.. 196.. done
>   Sub-merge = [1] 2807    5
>   Extract snp-gene pair = [1] 2613    2
>   Merge dim = [1] 2807  109
>   Write a CSV file: /data/20.08-Preterm_birth/summary_nearest.csv
>
>   GTEx dim = [1] 12796     7
>   Search biomaRt... 44.. parsing.. 44.. done
>   Merge dim = [1] 12796   113
>   Extract snp-eGene pair = [1] 815   2
>   Write a CSV file: /data/20.08-Preterm_birth/summary_gtex.csv
>
>   GWAS dim = [1] 2628   11
>   Merge dim = [1] 2616  115
>   Write a CSV file: /data/20.08-Preterm_birth/summary_gwas.csv
>
> Job done: 2020-08-15 04:47:28 for 20 sec

Upload latest image with new version

```CMD
# v5-2020-08-13
docker tag kisudsoe/postgwas:latest kisudsoe/postgwas:5
docker login
docker push kisudsoe/postgwas:latest
docker push kisudsoe/postgwas:5
```



## Venn analysis

### Enhancer SNPs from Roadmap & ENCODE

```CMD
docker run --rm ^
  -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" ^
  kisudsoe/postgwas Rscript /src/venn_analysis.r --fig ^
  --base /data/20.08-Preterm_birth/summary/snp_encode_tfbs_280.bed /data/20.08-Preterm_birth/summary/snp_union_roadmap_over_614.bed ^
  --out  /data/20.08-Preterm_birth
```

> Figure draw: /data/20.08-Preterm_birth/venn_snp_encode_tfbs_280_snp_union_roadmap_over_614.png



### Enhancer SNPs & RegulomeDB & GTEx eQTLs

```CMD
docker run --rm ^
  -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" ^
  kisudsoe/postgwas Rscript /src/venn_analysis.r --fig ^
  --base /data/20.08-Preterm_birth/Enhancer-TFBS_135.bed /data/20.08-Preterm_birth/summary/snp_regulome2b_34.bed /data/20.08-Preterm_birth/gtex_eqtl/snp_union_gtex_eqtl_815.bed ^
  --out  /data/20.08-Preterm_birth
```

>       original  fitted residuals regionError
> A           68  68.002    -0.002           0
> B           10  10.015    -0.015           0
> C          740 740.000     0.000           0
> A&B          4   3.958     0.042           0
> A&C         55  54.995     0.005           0
> B&C         12  11.984     0.016           0
> A&B&C        8   8.024    -0.024           0
>
> diagError: 0
> stress:    0
>
> Euler fit is done.
> Figure draw: /data/20.08-Preterm_birth/venn_Enhancer-TFBS_135_snp_regulome2b_34.png



## SNP-nexus for AA change annotation

Ref: https://www.snp-nexus.org/v4/7aa26925/



## SNPsea

Download and unzip ready-to-use program:

* Download; [util-snpsea](https://drive.google.com/file/d/1EMlk-hmK6mulFYIq7TYaOqZWyvRcP6MB/view?usp=sharing)

```bash
wget https://drive.google.com/file/d/1EMlk-hmK6mulFYIq7TYaOqZWyvRcP6MB/view?usp=sharing
mkdir snpsea
unzip util-snpsea.zip -d snpsea
```

Prepare GWAS file `20.08-Preterm_birth_2613.gwas`  by following below format:

```
<CHR> <POS> <SNP>
chr1  10000 rs100
```

Prepare bash file to run SNPsea. This is contents of `20.08-Preterm_birth.sh` file: 

```bash
#!/usr/bin/env bash

options=(
    --snps              20.08-Preterm_birth_2613.gwas
    --gene-matrix       FANTOM2014.gct.gz
    --gene-intervals    NCBIgenes2013.bed.gz
    --snp-intervals     TGP2011.bed.gz
    --null-snps         Lango2010.txt.gz
    --out               out/20.08-Preterm_birth-FANTOM
    --slop              10e3
    --threads           8
    --null-snpsets      0
    --min-observations  100
    --max-iterations    1e7
)

# Run SNPsea.
./bin/snpsea ${options[*]}

# Create a horizontal bar plot of condition p-values.
./bin/snpsea-barplot out/20.08-Preterm_birth-FANTOM


options=(
    --snps              20.08-Preterm_birth_2613.gwas
    --gene-matrix       GeneAtlas2004.gct.gz
    --gene-intervals    NCBIgenes2013.bed.gz
    --snp-intervals     TGP2011.bed.gz
    --null-snps         Lango2010.txt.gz
    --out               out/20.08-Preterm_birth-GA2004
    --slop              10e3
    --threads           8
    --null-snpsets      0
    --min-observations  100
    --max-iterations    1e7
)

# Run SNPsea.
./bin/snpsea ${options[*]}

# Create a horizontal bar plot of condition p-values.
./bin/snpsea-barplot out/20.08-Preterm_birth-GA2004


options=(
    --snps              20.08-Preterm_birth_2613.gwas
    --gene-matrix       GTEx_v8_gene_median_tpm.gct.gz
    --gene-intervals    Ensemblgenes2020.bed.gz
    --snp-intervals     TGP2011.bed.gz
    --null-snps         Lango2010.txt.gz
    --out               out/20.08-Preterm_birth-GTEx
    --slop              10e3
    --threads           8
    --null-snpsets      0
    --min-observations  100
    --max-iterations    1e7
)

# Run SNPsea.
./bin/snpsea ${options[*]}

# Create a horizontal bar plot of condition p-values.
./bin/snpsea-barplot out/20.08-Preterm_birth-GTEx
```

Run image for operating SNPsea:

```CMD
docker run -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" ^
  kisudsoe/postgwas /bin/sh -c "cd /data/util-snpsea && bash ./20.08-Preterm_birth.sh"
```

If error occurs for snpsea-barplot, do debugging and then;

Or change figure size for gtex at `bin/snpsea-barplot`;

```python
width = float(20) # 10 (GA, FANTOM) -> 20 (gtex)
top = int(60)     # 20 (GA, FANTOM) -> 60 (gtex)
```

Run for debugging:

```CMD
docker run -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" ^
  kisudsoe/postgwas /bin/sh -c ^
  "cd /data/util-snpsea && ./bin/snpsea-barplot out/20.08-Preterm_birth-FANTOM"

docker run -v "C:\Users\kisud\OneDrive\Suh's Lab\Postgwas_v3:/data" ^
  kisudsoe/postgwas /bin/sh -c ^
  "cd /data/util-snpsea && ./bin/snpsea-barplot out/20.08-Preterm_birth-GTEx"
```

